<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Differential abundance testing in microbiome data challenges both parametric and non-parametric statistical methods, due to its sparsity, high variability and compositional nature. Microbiome-specific statistical methods often assume classical distribution models or take into account compositional specifics. These produce results that range within the specificity vs sensitivity space in such a way that type I and type II error that are difficult to ascertain in real microbiome data when a single method is used. Recently, a consensus approach based on multiple differential abundance (DA) methods was recently suggested in order to increase robustness. With dar, you can use dplyr-like pipeable sequences of DA methods and then apply different consensus strategies. In this way we can obtain more reliable results in a fast, consistent and reproducible way.">
<title>Differential Abundance Analysis by Consensus â€¢ dar</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Differential Abundance Analysis by Consensus">
<meta property="og:description" content="Differential abundance testing in microbiome data challenges both parametric and non-parametric statistical methods, due to its sparsity, high variability and compositional nature. Microbiome-specific statistical methods often assume classical distribution models or take into account compositional specifics. These produce results that range within the specificity vs sensitivity space in such a way that type I and type II error that are difficult to ascertain in real microbiome data when a single method is used. Recently, a consensus approach based on multiple differential abundance (DA) methods was recently suggested in order to increase robustness. With dar, you can use dplyr-like pipeable sequences of DA methods and then apply different consensus strategies. In this way we can obtain more reliable results in a fast, consistent and reproducible way.">
<meta property="og:image" content="https://microbialgenomics-irsicaixaorg.github.io/dar/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">dar</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/dar.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/article.html">Workflow with real data</a>
    <a class="dropdown-item" href="articles/data_import.html">Converting Common Data Formats to Phyloseq and TreeSummarizedExperiment</a>
    <a class="dropdown-item" href="articles/filtering_subsetting.html">Filtering and Subsetting</a>
    <a class="dropdown-item" href="articles/import_export_recipes.html">Reproducibility in Microbiome Data Analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="navbar-nav">
<li><form class="form-inline" role="search">
<input type="search" class="form-control" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="dar-">dar 
<a class="anchor" aria-label="anchor" href="#dar-"></a>
</h1>
</div>
<!-- badges: start -->

<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Differential abundance testing in microbiome data challenges both parametric and non-parametric statistical methods, due to its sparsity, high variability and compositional nature. Microbiome-specific statistical methods often assume classical distribution models or take into account compositional specifics. These produce results that range within the specificity vs sensitivity space in such a way that type I and type II error are difficult to ascertain in real microbiome data when a single method is used. Recently, a consensus approach based on multiple differential abundance (DA) methods was recently suggested in order to increase robustness.</p>
<p>With dar, you can use dplyr-like pipeable sequences of DA methods and then apply different consensus strategies. In this way we can obtain more reliable results in a fast, consistent and reproducible way.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of dar from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MicrobialGenomics-IrsicaixaOrg/dar"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar" class="external-link">dar</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 methods overwritten by 'vegan':</span></span>
<span><span class="co">#&gt;   method         from      </span></span>
<span><span class="co">#&gt;   reorder.hclust seriation </span></span>
<span><span class="co">#&gt;   rev.hclust     dendextend</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metaHIV_phy"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Define recipe</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="reference/recipe.html">recipe</a></span><span class="op">(</span><span class="va">metaHIV_phy</span>, var_info <span class="op">=</span> <span class="st">"RiskGroup2"</span>, tax_info <span class="op">=</span> <span class="st">"Species"</span><span class="op">)</span> <span class="op"><a href="reference/grapes-greater-than-grapes.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/step_subset_taxa.html">step_subset_taxa</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="st">'Kingdom %in% c("Bacteria", "Archaea")'</span><span class="op">)</span> <span class="op"><a href="reference/grapes-greater-than-grapes.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/step_filter_taxa.html">step_filter_taxa</a></span><span class="op">(</span>.f <span class="op">=</span> <span class="st">"function(x) sum(x &gt; 0) &gt;= (0.03 * length(x))"</span><span class="op">)</span> <span class="op"><a href="reference/grapes-greater-than-grapes.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/step_metagenomeseq.html">step_metagenomeseq</a></span><span class="op">(</span>rm_zeros <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="reference/grapes-greater-than-grapes.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/step_maaslin.html">step_maaslin</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; â”€â”€ DAR Recipe â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â„¹ phyloseq object with 451 taxa and 156 samples </span></span>
<span><span class="co">#&gt;      â„¹ variable of interes RiskGroup2 (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      â„¹ taxonomic level Species </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â—‰ step_subset_taxa() id = subset_taxa__Suncake </span></span>
<span><span class="co">#&gt;      â—‰ step_filter_taxa() id = filter_taxa__Hot_water_crust_pastry </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â—‰ step_metagenomeseq() id = metagenomeseq__Crocetta_of_Caltanissetta </span></span>
<span><span class="co">#&gt;      â—‰ step_maaslin() id = maaslin__Tortita_negra</span></span>
<span></span>
<span><span class="co">## Prep recipe</span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/prep.html">prep</a></span><span class="op">(</span><span class="va">rec</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; â”€â”€ DAR Results â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â„¹ phyloseq object with 278 taxa and 156 samples </span></span>
<span><span class="co">#&gt;      â„¹ variable of interes RiskGroup2 (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      â„¹ taxonomic level Species </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      âœ” metagenomeseq__Crocetta_of_Caltanissetta diff_taxa = 236 </span></span>
<span><span class="co">#&gt;      âœ” maaslin__Tortita_negra diff_taxa = 146 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â„¹ 124 taxa are present in all tested methods</span></span>
<span></span>
<span><span class="co">## Consensus strategy</span></span>
<span><span class="va">n_methods</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/bake.html">bake</a></span><span class="op">(</span><span class="va">da_results</span>, count_cutoff <span class="op">=</span> <span class="va">n_methods</span><span class="op">)</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; â”€â”€ DAR Results â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â„¹ phyloseq object with 278 taxa and 156 samples </span></span>
<span><span class="co">#&gt;      â„¹ variable of interes RiskGroup2 (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      â„¹ taxonomic level Species </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      âœ” metagenomeseq__Crocetta_of_Caltanissetta diff_taxa = 236 </span></span>
<span><span class="co">#&gt;      âœ” maaslin__Tortita_negra diff_taxa = 146 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â„¹ 124 taxa are present in all tested methods </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Bakes:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      â—‰ 1 -&gt; count_cutoff: 2, weights: NULL, exclude: NULL, id: bake__KÃ¼rtÅ‘skalÃ¡cs</span></span>
<span></span>
<span><span class="co">## Results</span></span>
<span><span class="fu"><a href="reference/cool.html">cool</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; â„¹ Bake for count_cutoff = 2</span></span>
<span><span class="co">#&gt; # A tibble: 124 Ã— 2</span></span>
<span><span class="co">#&gt;    taxa_id taxa                   </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;                  </span></span>
<span><span class="co">#&gt;  1 Otu_63  Bacteroides_plebeius   </span></span>
<span><span class="co">#&gt;  2 Otu_216 Clostridium_sp_CAG_632 </span></span>
<span><span class="co">#&gt;  3 Otu_441 Brachyspira_sp_CAG_700 </span></span>
<span><span class="co">#&gt;  4 Otu_108 Prevotella_sp_CAG_520  </span></span>
<span><span class="co">#&gt;  5 Otu_257 Butyrivibrio_sp_CAG_318</span></span>
<span><span class="co">#&gt;  6 Otu_104 Prevotella_sp_CAG_1092 </span></span>
<span><span class="co">#&gt;  7 Otu_69  Bacteroides_sp_CAG_530 </span></span>
<span><span class="co">#&gt;  8 Otu_102 Prevotella_sp_AM42_24  </span></span>
<span><span class="co">#&gt;  9 Otu_159 Lactobacillus_ruminis  </span></span>
<span><span class="co">#&gt; 10 Otu_117 Alistipes_inops        </span></span>
<span><span class="co">#&gt; # â„¹ 114 more rows</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<ul>
<li><p>If you think you have encountered a bug, please <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/issues" class="external-link">submit an issue</a>.</p></li>
<li><p>Either way, learn how to create and share a <a href="https://reprex.tidyverse.org/articles/articles/learn-reprex.html" class="external-link">reprex</a> (a minimal, reproducible example), to clearly communicate about your code.</p></li>
<li><p>Working on your first Pull Request? You can learn how from this <em>free</em> series <a href="https://kcd.im/pull-request" class="external-link">How to Contribute to an Open Source Project on GitHub</a></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the dar project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/dar" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing dar</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Francesc Catala-Moll <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-2354-8648" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/actions" class="external-link"><img src="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/workflows/R-CMD-check-bioc/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/MicrobialGenomics-IrsicaixaOrg/dar?branch=devel" class="external-link"><img src="https://codecov.io/gh/MicrobialGenomics-IrsicaixaOrg/dar/branch/devel/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://makeapullrequest.com" class="external-link"><img src="https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square" alt="PRs Welcome"></a></li>
<li><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/issues" class="external-link"><img src="https://img.shields.io/github/issues/MicrobialGenomics-IrsicaixaOrg/dar" alt="GitHub issues"></a></li>
<li><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/MicrobialGenomics-IrsicaixaOrg/dar" alt="GitHub pulls"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Francesc Catala-Moll.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
