<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ dar</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="https://microbialgenomics-irsicaixaorg.github.io/dar/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">dar</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/dar.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/article.html">Workflow with real data</a></li>
    <li><a class="dropdown-item" href="../articles/data_import.html">Converting Common Data Formats to Phyloseq and TreeSummarizedExperiment</a></li>
    <li><a class="dropdown-item" href="../articles/filtering_subsetting.html">Filtering and Subsetting</a></li>
    <li><a class="dropdown-item" href="../articles/import_export_recipes.html">Reproducibility in Microbiome Data Analysis</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="basic-functions">Basic Functions<a class="anchor" aria-label="anchor" href="#basic-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="dar-package.html">dar</a></code> <code><a href="dar-package.html">dar-package</a></code>

        </dt>
        <dd>dar: Differential Abundance Analysis by Consensus</dd>
      </dl><dl><dt>

          <code><a href="recipe.html">recipe()</a></code>

        </dt>
        <dd>Create a Recipe for preprocessing data</dd>
      </dl><dl><dt>

          <code><a href="prep.html">prep()</a></code>

        </dt>
        <dd>Performs all the steps defined in a Recipe</dd>
      </dl><dl><dt>

          <code><a href="bake.html">bake()</a></code>

        </dt>
        <dd>Define consensus strategies from a Recipe</dd>
      </dl><dl><dt>

          <code><a href="cool.html">cool()</a></code>

        </dt>
        <dd>Extract results from defined bake</dd>
      </dl></div><div class="section level2">
      <h2 id="step-functions-data-preprocessing">Step Functions - Data Preprocessing<a class="anchor" aria-label="anchor" href="#step-functions-data-preprocessing"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="step_filter_taxa.html">step_filter_taxa()</a></code>

        </dt>
        <dd>Filter taxa based on across-sample OTU abundance criteria</dd>
      </dl><dl><dt>

          <code><a href="step_filter_by_prevalence.html">step_filter_by_prevalence()</a></code>

        </dt>
        <dd>Filter taxa by prevalence</dd>
      </dl><dl><dt>

          <code><a href="step_filter_by_abundance.html">step_filter_by_abundance()</a></code>

        </dt>
        <dd>Filter taxa by abundance</dd>
      </dl><dl><dt>

          <code><a href="step_filter_by_rarity.html">step_filter_by_rarity()</a></code>

        </dt>
        <dd>Filter taxa by rarity</dd>
      </dl><dl><dt>

          <code><a href="step_filter_by_variance.html">step_filter_by_variance()</a></code>

        </dt>
        <dd>Filter taxa by variance</dd>
      </dl><dl><dt>

          <code><a href="step_subset_taxa.html">step_subset_taxa()</a></code>

        </dt>
        <dd>Subset taxa by taxonomic level</dd>
      </dl><dl><dt>

          <code><a href="step_rarefaction.html">step_rarefaction()</a></code>

        </dt>
        <dd>Resample an OTU table such that all samples have the same library size.</dd>
      </dl></div><div class="section level2">
      <h2 id="step-functions-differential-abundance-methods">Step Functions - Differential Abundance Methods<a class="anchor" aria-label="anchor" href="#step-functions-differential-abundance-methods"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="step_aldex.html">step_aldex()</a></code>

        </dt>
        <dd>ALDEx2 analysis</dd>
      </dl><dl><dt>

          <code><a href="step_ancom.html">step_ancom()</a></code>

        </dt>
        <dd>ANCOM analysis</dd>
      </dl><dl><dt>

          <code><a href="step_corncob.html">step_corncob()</a></code>

        </dt>
        <dd>corncob analysis</dd>
      </dl><dl><dt>

          <code><a href="step_deseq.html">step_deseq()</a></code>

        </dt>
        <dd>DESeq2 analysis</dd>
      </dl><dl><dt>

          <code><a href="step_lefse.html">step_lefse()</a></code>

        </dt>
        <dd>lefse analysis</dd>
      </dl><dl><dt>

          <code><a href="step_maaslin.html">step_maaslin()</a></code>

        </dt>
        <dd>MaAsLin2 analysis</dd>
      </dl><dl><dt>

          <code><a href="step_metagenomeseq.html">step_metagenomeseq()</a></code>

        </dt>
        <dd>MetagenomeSeq analysis</dd>
      </dl><dl><dt>

          <code><a href="step_wilcox.html">step_wilcox()</a></code>

        </dt>
        <dd>Wilcox analysis</dd>
      </dl></div><div class="section level2">
      <h2 id="exploration">Exploration<a class="anchor" aria-label="anchor" href="#exploration"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="abundance_plt.html">abundance_plt()</a></code>

        </dt>
        <dd>Abundance boxplot</dd>
      </dl><dl><dt>

          <code><a href="corr_heatmap.html">corr_heatmap()</a></code>

        </dt>
        <dd>Plot otuput of the <code>overlap_df</code> function as a heatmap.</dd>
      </dl><dl><dt>

          <code><a href="exclusion_plt.html">exclusion_plt()</a></code>

        </dt>
        <dd>Plot the number of shared DA OTUs between methods.</dd>
      </dl><dl><dt>

          <code><a href="find_intersections.html">find_intersections()</a></code>

        </dt>
        <dd>Finds common OTU between method results</dd>
      </dl><dl><dt>

          <code><a href="intersection_df.html">intersection_df()</a></code>

        </dt>
        <dd>Returns data.frame with OTU intersection between methods</dd>
      </dl><dl><dt>

          <code><a href="intersection_plt.html">intersection_plt()</a></code>

        </dt>
        <dd>Plot results using UpSet plot</dd>
      </dl><dl><dt>

          <code><a href="mutual_plt.html">mutual_plt()</a></code>

        </dt>
        <dd>Mutual finding plot</dd>
      </dl><dl><dt>

          <code><a href="overlap_df.html">overlap_df()</a></code>

        </dt>
        <dd>Overlap of significant OTUs between tested methods.</dd>
      </dl></div><div class="section level2">
      <h2 id="recipe-getters-and-adders">Recipe Getters and Adders<a class="anchor" aria-label="anchor" href="#recipe-getters-and-adders"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="add_tax.html">add_tax()</a></code>

        </dt>
        <dd>Adds taxonomic level of interest in the Recipe.</dd>
      </dl><dl><dt>

          <code><a href="add_var.html">add_var()</a></code>

        </dt>
        <dd>Adds variable of interest to the Recipe</dd>
      </dl><dl><dt>

          <code><a href="get_phy.html">get_phy()</a></code>

        </dt>
        <dd>Returns phyloseq from Recipe-class object</dd>
      </dl><dl><dt>

          <code><a href="get_tax.html">get_tax()</a></code>

        </dt>
        <dd>Returns tax_info from Recipe-class object</dd>
      </dl><dl><dt>

          <code><a href="get_var.html">get_var()</a></code>

        </dt>
        <dd>Returns var_info from Recipe-class object</dd>
      </dl><dl><dt>

          <code><a href="otu_table.html">otu_table()</a></code>

        </dt>
        <dd>Extracts otu_table from phyloseq inside a Recipe</dd>
      </dl><dl><dt>

          <code><a href="phy_qc.html">phy_qc()</a></code>

        </dt>
        <dd>Phyloseq Quality Control Metrics</dd>
      </dl><dl><dt>

          <code><a href="sample_data.html">sample_data()</a></code>

        </dt>
        <dd>Extracts sample_data from phyloseq inside a Recipe</dd>
      </dl><dl><dt>

          <code><a href="steps_ids.html">steps_ids()</a></code>

        </dt>
        <dd>Get step_ids from recipe</dd>
      </dl><dl><dt>

          <code><a href="tax_table.html">tax_table()</a></code>

        </dt>
        <dd>Extracts tax_table from phyloseq inside a Recipe</dd>
      </dl><dl><dt>

          <code><a href="required_deps.html">required_deps()</a></code>

        </dt>
        <dd>Returns required pakcages for Recipe object</dd>
      </dl><dl><dt>

          <code><a href="zero_otu.html">zero_otu()</a></code>

        </dt>
        <dd>Extract outs with all 0 values in at least on level of the variable</dd>
      </dl></div><div class="section level2">
      <h2 id="import-and-export">Import and Export<a class="anchor" aria-label="anchor" href="#import-and-export"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="export_steps.html">export_steps()</a></code>

        </dt>
        <dd>Export step parameters as json.</dd>
      </dl><dl><dt>

          <code><a href="import_steps.html">import_steps()</a></code>

        </dt>
        <dd>Import steps from json file</dd>
      </dl><dl><dt>

          <code><a href="read_data.html">read_data()</a></code> <code><a href="read_data.html">validate_otu()</a></code> <code><a href="read_data.html">validate_sample_data()</a></code> <code><a href="read_data.html">validate_tax_table()</a></code> <code><a href="read_data.html">validate_phyloseq()</a></code> <code><a href="read_data.html">read_phyloseq()</a></code> <code><a href="read_data.html">read_file()</a></code>

        </dt>
        <dd>Loads Phyloseq data</dd>
      </dl></div><div class="section level2">
      <h2 id="other-functions">Other functions<a class="anchor" aria-label="anchor" href="#other-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="contains_rarefaction.html">contains_rarefaction()</a></code>

        </dt>
        <dd>Checks if Recipe contains a rarefaction step</dd>
      </dl><dl><dt>

          <code><a href="rand_id.html">rand_id()</a></code>

        </dt>
        <dd>Make a random identification field for steps</dd>
      </dl><dl><dt>

          <code><a href="PrepRecipe-class.html">PrepRecipe-class</a></code>

        </dt>
        <dd>PrepRecipe-class object</dd>
      </dl><dl><dt>

          <code><a href="Recipe-class.html">show(<i>&lt;PrepRecipe&gt;</i>)</a></code>

        </dt>
        <dd>Recipe-class object</dd>
      </dl></div><div class="section level2">
      <h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="metaHIV_phy.html">metaHIV_phy</a></code>

        </dt>
        <dd>Phyloseq object from metaHIV project</dd>
      </dl><dl><dt>

          <code><a href="pastry_df.html">pastry_df</a></code>

        </dt>
        <dd>Pastery data for step id generation</dd>
      </dl><dl><dt>

          <code><a href="test_rec.html">test_rec</a></code>

        </dt>
        <dd>Recipe for metaHIV_phy data</dd>
      </dl><dl><dt>

          <code><a href="test_prep_rec.html">test_prep_rec</a></code>

        </dt>
        <dd>PrepRecipe for metaHIV_phy data</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Francesc Catala-Moll.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer></div>





  </body></html>

