<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Determine taxa whose absolute abundances, per unit volume, of the ecosystem
(e.g., gut) are significantly different with changes in the covariate of
interest (e.g., group). The current version of ancombc2 function implements
Analysis of Compositions of Microbiomes with Bias Correction (ANCOM-BC2) in
cross-sectional and repeated measurements data. In addition to the two-group
comparison, ANCOM-BC2 also supports testing for continuous covariates and
multi-group comparisons, including the global test, pairwise directional
test, Dunnett's type of test, and trend test."><title>ANCOM analysis — step_ancom • dar</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ANCOM analysis — step_ancom"><meta property="og:description" content="Determine taxa whose absolute abundances, per unit volume, of the ecosystem
(e.g., gut) are significantly different with changes in the covariate of
interest (e.g., group). The current version of ancombc2 function implements
Analysis of Compositions of Microbiomes with Bias Correction (ANCOM-BC2) in
cross-sectional and repeated measurements data. In addition to the two-group
comparison, ANCOM-BC2 also supports testing for continuous covariates and
multi-group comparisons, including the global test, pairwise directional
test, Dunnett's type of test, and trend test."><meta property="og:image" content="https://microbialgenomics-irsicaixaorg.github.io/dar/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dar</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/dar.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/article.html">Workflow with real data</a>
    <a class="dropdown-item" href="../articles/data_import.html">Data Import</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ANCOM analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/blob/HEAD/R/ancom.R" class="external-link"><code>R/ancom.R</code></a></small>
      <div class="d-none name"><code>step_ancom.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Determine taxa whose absolute abundances, per unit volume, of the ecosystem
(e.g., gut) are significantly different with changes in the covariate of
interest (e.g., group). The current version of ancombc2 function implements
Analysis of Compositions of Microbiomes with Bias Correction (ANCOM-BC2) in
cross-sectional and repeated measurements data. In addition to the two-group
comparison, ANCOM-BC2 also supports testing for continuous covariates and
multi-group comparisons, including the global test, pairwise directional
test, Dunnett's type of test, and trend test.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">step_ancom</span><span class="op">(</span></span>
<span>  <span class="va">rec</span>,</span>
<span>  fix_formula <span class="op">=</span> <span class="fu"><a href="get_var.html">get_var</a></span><span class="op">(</span><span class="va">rec</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  rand_formula <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  p_adj_method <span class="op">=</span> <span class="st">"holm"</span>,</span>
<span>  pseudo <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  pseudo_sens <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  prv_cut <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  lib_cut <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  s0_perc <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  group <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  struc_zero <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  neg_lb <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  n_cl <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  global <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  pairwise <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dunnet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  trend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  rarefy <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  id <span class="op">=</span> <span class="fu"><a href="rand_id.html">rand_id</a></span><span class="op">(</span><span class="st">"ancom"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for recipe</span></span>
<span><span class="fu">step_ancom</span><span class="op">(</span></span>
<span>  <span class="va">rec</span>,</span>
<span>  fix_formula <span class="op">=</span> <span class="fu"><a href="get_var.html">get_var</a></span><span class="op">(</span><span class="va">rec</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  rand_formula <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  p_adj_method <span class="op">=</span> <span class="st">"holm"</span>,</span>
<span>  pseudo <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  pseudo_sens <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  prv_cut <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  lib_cut <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  s0_perc <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  group <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  struc_zero <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  neg_lb <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  n_cl <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  global <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  pairwise <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dunnet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  trend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  rarefy <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  id <span class="op">=</span> <span class="fu"><a href="rand_id.html">rand_id</a></span><span class="op">(</span><span class="st">"ancom"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>rec</dt>
<dd><p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p></dd>


<dt>fix_formula</dt>
<dd><p>the character string expresses how the microbial absolute
abundances for each taxon depend on the fixed effects in metadata.</p></dd>


<dt>rand_formula</dt>
<dd><p>the character string expresses how the microbial absolute
abundances for each taxon depend on the random effects in metadata.
ANCOM-BC2 follows the lmerTest package in formulating the random effects.
See ?lmerTest::lmer for more details. Default is NULL.</p></dd>


<dt>p_adj_method</dt>
<dd><p>character. method to adjust p-values. Default is "holm".
Options include "holm", "hochberg", "hommel", "bonferroni", "BH", "BY",
"fdr", "none". See ?stats::p.adjust for more details.</p></dd>


<dt>pseudo</dt>
<dd><p>numeric. Add pseudo-counts to the data. Default is 0 (no
pseudo-count addition).</p></dd>


<dt>pseudo_sens</dt>
<dd><p>ogical. Whether to perform the sensitivity analysis to the
pseudo-count addition. Default is TRUE. See Details for a more
comprehensive discussion on this sensitivity analysis.</p></dd>


<dt>prv_cut</dt>
<dd><p>a numerical fraction between 0 and 1. Taxa with prevalences
less than prv_cut will be excluded in the analysis. For instance, suppose
there are 100 samples, if a taxon has nonzero counts presented in less than
10 samples, it will not be further analyzed. Default is 0.10.</p></dd>


<dt>lib_cut</dt>
<dd><p>a numerical threshold for filtering samples based on library
sizes. Samples with library sizes less than lib_cut will be excluded in the
analysis. Default is 0, i.e. do not discard any sample.</p></dd>


<dt>s0_perc</dt>
<dd><p>a numerical fraction between 0 and 1. Inspired by Significance
Analysis of Microarrays (SAM) methodology, a small positive constant is
added to the denominator of ANCOM-BC2 test statistic corresponding to each
taxon to avoid the significance due to extremely small standard errors,
especially for rare taxa. This small positive constant is chosen as
s0_perc-th percentile of standard error values for each fixed effect.
Default is 0.05 (5th percentile).</p></dd>


<dt>group</dt>
<dd><p>character. The name of the group variable in metadata. group
should be discrete. Specifying group is required for detecting structural
zeros and performing multi-group comparisons (global test, pairwise
directional test, Dunnett's type of test, and trend test). Default is NULL.
If the group of interest contains only two categories, leave it as NULL.</p></dd>


<dt>struc_zero</dt>
<dd><p>logical. Whether to detect structural zeros based on group.
Default is FALSE. See Details for a more comprehensive discussion on
structural zeros.</p></dd>


<dt>neg_lb</dt>
<dd><p>logical. Whether to classify a taxon as a structural zero using
its asymptotic lower bound. Default is FALSE.</p></dd>


<dt>alpha</dt>
<dd><p>numeric. Level of significance. Default is 0.05.</p></dd>


<dt>n_cl</dt>
<dd><p>numeric. The number of nodes to be forked. For details, see
?parallel::makeCluster. Default is 1 (no parallel computing).</p></dd>


<dt>verbose</dt>
<dd><p>logical. Whether to generate verbose output during the
ANCOM-BC2 fitting process. Default is FALSE.</p></dd>


<dt>global</dt>
<dd><p>logical. Whether to perform the global test. Default is FALSE.</p></dd>


<dt>pairwise</dt>
<dd><p>logical. Whether to perform the pairwise directional test.
Default is FALSE.</p></dd>


<dt>dunnet</dt>
<dd><p>logical. Whether to perform the Dunnett's type of test. Default
is FALSE.</p></dd>


<dt>trend</dt>
<dd><p>logical. Whether to perform trend test. Default is FALSE.</p></dd>


<dt>rarefy</dt>
<dd><p>Boolean indicating if OTU counts must be rarefyed. This
rarefaction uses the standard R sample function to resample from the
abundance values in the otu_table component of the first argument, physeq.
Often one of the major goals of this procedure is to achieve parity in
total number of counts between samples, as an alternative to other formal
normalization procedures, which is why a single value for the sample.size
is expected.</p></dd>


<dt>id</dt>
<dd><p>A character string that is unique to this step to identify it.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>An object of class <code>recipe</code></p>


    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Diff taxa steps: 
<code><a href="step_aldex.html">step_aldex</a>()</code>,
<code><a href="step_corncob.html">step_corncob</a>()</code>,
<code><a href="step_deseq.html">step_deseq</a>()</code>,
<code><a href="step_lefse.html">step_lefse</a>()</code>,
<code><a href="step_maaslin.html">step_maaslin</a>()</code>,
<code><a href="step_metagenomeseq.html">step_metagenomeseq</a>()</code>,
<code><a href="step_wilcox.html">step_wilcox</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metaHIV_phy</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Init recipe</span></span></span>
<span class="r-in"><span><span class="va">rec</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="recipe.html">recipe</a></span><span class="op">(</span><span class="va">metaHIV_phy</span>, <span class="st">"RiskGroup2"</span>, <span class="st">"Species"</span><span class="op">)</span> <span class="op"><a href="pipe.html">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="step_subset_taxa.html">step_subset_taxa</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="st">'Kingdom %in% c("Bacteria", "Archaea")'</span><span class="op">)</span> <span class="op"><a href="pipe.html">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="step_filter_taxa.html">step_filter_taxa</a></span><span class="op">(</span>.f <span class="op">=</span> <span class="st">"function(x) sum(x &gt; 0) &gt;= (0.4 * length(x))"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">rec</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Preporcessing steps:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Kouign_amann</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">◉</span> step_filter_taxa() <span style="color: #555555;">id = filter_taxa__Sfenj</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DA steps:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Define step with default parameters and prep</span></span></span>
<span class="r-in"><span><span class="va">rec</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>  <span class="fu">step_ancom</span><span class="op">(</span><span class="va">rec</span><span class="op">)</span> <span class="op"><a href="pipe.html">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="prep.html">prep</a></span><span class="op">(</span>parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Registered S3 methods overwritten by 'proxy':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   method               from    </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.registry_field registry</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.registry_entry registry</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">rec</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ── <span style="color: #0000BB;">DAR Results</span> ─────────────────────────────────────────────────────────────────</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">76</span> taxa and <span style="color: #0000BB;">156</span> samples </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Results:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #00BB00;">✔</span> ancom__Rugelach <span style="color: #555555;">diff_taxa = 28</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> 28 taxa are present in all tested methods </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Wearing rarefaction only for this step</span></span></span>
<span class="r-in"><span><span class="va">rec</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="recipe.html">recipe</a></span><span class="op">(</span><span class="va">metaHIV_phy</span>, <span class="st">"RiskGroup2"</span>, <span class="st">"Species"</span><span class="op">)</span> <span class="op"><a href="pipe.html">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu">step_ancom</span><span class="op">(</span><span class="va">rec</span>, rarefy <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">rec</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Preporcessing steps:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DA steps:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      <span style="color: #0000BB;">◉</span> step_ancom() <span style="color: #555555;">id = ancom__Welsh_cake</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Francesc Catala-Moll, Marc Noguera-Julian.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

