<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="dar">
<title>Workflow with real data • dar</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Workflow with real data">
<meta property="og:description" content="dar">
<meta property="og:image" content="https://microbialgenomics-irsicaixaorg.github.io/dar/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dar</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.13</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/dar.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/article.html">Workflow with real data</a>
    <a class="dropdown-item" href="../articles/data_import.html">Converting Common Data Formats to Phyloseq and TreeSummarizedExperiment</a>
    <a class="dropdown-item" href="../articles/filtering_subsetting.html">Filtering and Subsetting</a>
    <a class="dropdown-item" href="../articles/import_export_recipes.html">Reproducibility in Microbiome Data Analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="navbar-nav">
<li><form class="form-inline" role="search">
<input type="search" class="form-control" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<link href="article_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="article_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="article_files/plotly-binding-4.10.4/plotly.js"></script><script src="article_files/typedarray-0.1/typedarray.min.js"></script><link href="article_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="article_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><link href="article_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="article_files/plotly-main-2.11.1/plotly-latest.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Workflow with real data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/blob/HEAD/vignettes/article.Rmd" class="external-link"><code>vignettes/article.Rmd</code></a></small>
      <div class="d-none name"><code>article.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<p>To illustrate the functionality of the <code>dar</code> package, this
study will use the data set from (Noguera-Julian, M., et al. 2016). The
authors of this study found that men who have sex with men (MSM)
predominantly belonged to the Prevotella-rich enterotype whereas most
non-MSM subjects were enriched in Bacteroides, independently of HIV-1
status. This result highlights the potential impact of sexual
orientation on the gut microbiome and emphasizes the importance of
controlling for such variables in microbiome research. Using the
<code>dar</code> package, we will conduct a differential abundance
analysis to further explore this finding and uncover potential microbial
biomarkers associated with this specific population.</p>
<div class="section level2">
<h2 id="load-dar-package-and-data">Load dar package and data<a class="anchor" aria-label="anchor" href="#load-dar-package-and-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar" class="external-link">dar</a></span><span class="op">)</span></span>
<span><span class="co"># suppressPackageStartupMessages(library(plotly))</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metaHIV_phy"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaHIV_phy</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 451 taxa and 156 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 156 samples by 3 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 451 taxa by 7 taxonomic ranks ]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="recipe-initialization">Recipe initialization<a class="anchor" aria-label="anchor" href="#recipe-initialization"></a>
</h2>
<p>To begin the analysis process with the <code>dar</code> package, the
first step is to initialize a <code>Recipe</code> object, which is an S4
class. This recipe object serves as a blueprint for the data preparation
steps required for the differential abundance analysis. The
initialization of the recipe object is done through the function
<code><a href="../reference/recipe.html">recipe()</a></code>, which takes as inputs a <code>phyloseq</code> or
<code>TreeSummarizedExperiment</code> (TSE) object, the name of the
categorical variable of interest and the taxonomic level at which the
differential abundance analyses are to be performed. As previously
mentioned, we will use the data set from (Noguera-Julian, M., et
al. 2016) and the variable of interest “RiskGroup2” containing the
categories: men who have sex with men (msm), non-MSM (hts) and people
who inject drugs (pwid) and we will perform the analysis at the species
level.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Recipe initialization</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/recipe.html">recipe</a></span><span class="op">(</span><span class="va">metaHIV_phy</span>, var_info <span class="op">=</span> <span class="st">"RiskGroup2"</span>, tax_info <span class="op">=</span> <span class="st">"Species"</span><span class="op">)</span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="recipe-qc-and-preprocessing-steps-definition">Recipe QC and preprocessing steps definition<a class="anchor" aria-label="anchor" href="#recipe-qc-and-preprocessing-steps-definition"></a>
</h2>
<p>Once the recipe object has been initialized, the next step is to
populate it with steps. Steps are the methods that will be applied to
the data stored in the recipe. There are two types of steps:
preprocessing (prepro) and differential abundance (da) steps. Initially,
we will focus on the prepro steps which are used to modify the data
loaded into the recipe, which will then be used for the da steps. The
‘dar’ package includes 3 main preprocessing functionalities:
<code>step_subset_taxa</code>, which is used for subsetting columns and
values in the taxon table connected to the phyloseq object,
<code>step_filter_taxa</code>, which is used to filter the OTUs, and
<code>step_rarefaction</code>, which is used to resample the OTU table
to ensure that all samples have the same library size. These
functionalities allow for a high level of flexibility and customization
in the data preparation process before performing the differential
abundance analysis.</p>
<p>The <code>dar</code> package provides convenient wrappers for the
<code>step_filter_taxa</code> function, designed to filter Operational
Taxonomic Units (OTUs) based on specific criteria: prevalence, variance,
abundance, and rarity.</p>
<ul>
<li>
<code>step_filter_by_prevalence</code>: Filters OTUs according to
the number of samples in which the OTU appears.</li>
<li>
<code>step_filter_by_variance</code>: Filters OTUs based on the
variance of the OTU’s presence across samples.</li>
<li>
<code>step_filter_by_abundance</code>: Filters OTUs according to the
OTU’s abundance across samples.</li>
<li>
<code>step_filter_by_rarity</code>: Filters OTUs based on the rarity
of the OTU across samples.</li>
</ul>
<p>In addition to the preprocessing steps, the <code>dar</code> package
also incorporates the function <code>phy_qc</code> which returns a table
with a set of metrics that allow for informed decisions to be made about
the data preprocessing that will be done. In our case, we decided to use
the step_subset_taxa function to retain only those observations
annotated within the realm of Bacteria and Archaea. We also used the
<code>step_filter_by_prevalence</code> function to retain only those
OTUs with at least 1% of the samples with values greater than 0. This
approach ensured that we were working with a high-quality, informative
subset of the data, which improved the overall accuracy and reliability
of the differential abundance analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## QC </span></span>
<span><span class="fu"><a href="../reference/phy_qc.html">phy_qc</a></span><span class="op">(</span><span class="va">rec</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 10</span></span></span>
<span><span class="co">#&gt;   var_levels     n n_zero pct_zero pct_all_zero pct_singletons pct_doubletons</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> all        <span style="text-decoration: underline;">70</span>356  <span style="text-decoration: underline;">57</span>632     81.9          0             20.6           8.87</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> hts        <span style="text-decoration: underline;">18</span>491  <span style="text-decoration: underline;">15</span>108     81.7         24.2           22.8           8.43</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> msm        <span style="text-decoration: underline;">45</span>100  <span style="text-decoration: underline;">37</span>019     82.1         16.0           20.2           9.53</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> pwid        <span style="text-decoration: underline;">6</span>765   <span style="text-decoration: underline;">5</span>505     81.4         41.2           16.6           9.31</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: count_mean &lt;dbl&gt;, count_min &lt;dbl&gt;, count_max &lt;dbl&gt;</span></span></span>
<span></span>
<span><span class="co">## Adding prepro steps</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">rec</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/step_subset_taxa.html">step_subset_taxa</a></span><span class="op">(</span>tax_level <span class="op">=</span> <span class="st">"Kingdom"</span>, taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bacteria"</span>, <span class="st">"Archaea"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/step_filter_by_prevalence.html">step_filter_by_prevalence</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Puff_pastry</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_filter_by_prevalence() <span style="color: #555555;">id = filter_by_prevalence__Paris–Brest</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-differential-analysis-da-steps">Define Differential Analysis (DA) steps<a class="anchor" aria-label="anchor" href="#define-differential-analysis-da-steps"></a>
</h2>
<p>Once data is preprocessed and cleaned, the next step is to add the da
steps. The <code>dar</code> package incorporates multiple methods to
analyze the data, including: ALDEx2, ANCOM-BC, corncob, DESeq2, Lefse,
MAaslin2, MetagenomeSeq, and Wilcox. These methods provide a range of
options for uncovering potential microbial biomarkers associated with
the variable of interest. To ensure consistency across methods, we
decided not to use default parameters, but to set the
<code>min_prevalence</code> parameter to 0 for MAaslin2, and the
<code>rm_zeros</code> parameter to 0.01 for MetagenomeSeq, since it was
observed that the pct_all_zeros value was not equal to 0 in some levels
of the categorical variable in the results of <code><a href="../reference/phy_qc.html">phy_qc()</a></code>.
This approach ensured that the analysis was consistent across all
methods and that the results were interpretable.</p>
<p>Note: to reduce computation time, in this example we will only use
the metagenomeSeq and MAaslin2 methods, that are the fastest ones.
However, we recommend using all the methods available in the package to
ensure a more robust analysis.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## DA steps definition</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span> <span class="va">rec</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/step_metagenomeseq.html">step_metagenomeseq</a></span><span class="op">(</span>rm_zeros <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/step_maaslin.html">step_maaslin</a></span><span class="op">(</span>min_prevalence <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Puff_pastry</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_filter_by_prevalence() <span style="color: #555555;">id = filter_by_prevalence__Paris–Brest</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_metagenomeseq() <span style="color: #555555;">id = metagenomeseq__Spina_santa</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_maaslin() <span style="color: #555555;">id = maaslin__Dutch_Baby_Pancake</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prep-recipe">Prep recipe<a class="anchor" aria-label="anchor" href="#prep-recipe"></a>
</h2>
<p>Once the recipe has been defined, the next step is to execute all the
steps defined in the recipe. This is done through the function
<code><a href="../reference/prep.html">prep()</a></code>. Internally, it first executes the preprocessing
steps, which modify the phyloseq object stored in the recipe. Then,
using the modified phyloseq, it executes each of the defined
differential abundance methods. To speed up the execution time, the
<code><a href="../reference/prep.html">prep()</a></code> function includes the option to run in parallel. The
resulting object has class <code>PrepRecipe</code> and when printed in
the terminal, it displays the number of taxa detected as significant in
each of the methods and also the total number of taxa shared across all
methods. This allows for a provisional overview of the results and a
comparison between methods.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Execute in parallel</span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep.html">prep</a></span><span class="op">(</span><span class="va">rec</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in sqrt(out$s2.post): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(out$s2.post): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(out$s2.post): NaNs produced</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Results</span> ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">355</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> metagenomeseq__Spina_santa <span style="color: #555555;">diff_taxa = 294</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> maaslin__Dutch_Baby_Pancake <span style="color: #555555;">diff_taxa = 235</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> 210 taxa are present in all tested methods</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="default-results-extraction">Default results extraction<a class="anchor" aria-label="anchor" href="#default-results-extraction"></a>
</h2>
<p>At this point, we could extract the taxa shared across all methods
using the function <code><a href="../reference/bake.html">bake()</a></code> to define a default consensus
strategy and then <code><a href="../reference/cool.html">cool()</a></code> to extract the results.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Default DA taxa results</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/bake.html">bake</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/cool.html">cool</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 210 × 2</span></span></span>
<span><span class="co">#&gt;    taxa_id taxa                   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Otu_70  Bacteroides_sp_CAG_598 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Otu_73  Bacteroides_sp_D2      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Otu_369 Dialister_sp_CAG_357   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Otu_121 Alistipes_sp_An31A     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Otu_63  Bacteroides_plebeius   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Otu_216 Clostridium_sp_CAG_632 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Otu_257 Butyrivibrio_sp_CAG_318</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Otu_137 Enterococcus_avium     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Otu_49  Bacteroides_coprocola  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Otu_441 Brachyspira_sp_CAG_700 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 200 more rows</span></span></span></code></pre></div>
<p>However, <code>dar</code> allows for complex consensus strategies
based on the obtained results. To that end, the user has access to
different functions to graphically represent different types of
information. This feature allows for a more in-depth analysis of the
results and a better understanding of the underlying patterns in the
data.</p>
</div>
<div class="section level2">
<h2 id="exploration-for-consensus-strategie-definition">Exploration for consensus strategie definition<a class="anchor" aria-label="anchor" href="#exploration-for-consensus-strategie-definition"></a>
</h2>
<p>For example, <code><a href="../reference/intersection_plt.html">intersection_plt()</a></code> gives an overview of the
overlaps between methods by creating an upSet plot. In our case, this
function has shown that 210 taxa are shared across all the methods
used.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Intersection plot </span></span>
<span><span class="fu"><a href="../reference/intersection_plt.html">intersection_plt</a></span><span class="op">(</span><span class="va">da_results</span>, ordered_by <span class="op">=</span> <span class="st">"degree"</span>, font_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;"></p>
<p>In addition to the <code><a href="../reference/intersection_plt.html">intersection_plt()</a></code> function,
<code>dar</code> also has the function <code><a href="../reference/exclusion_plt.html">exclusion_plt()</a></code>
which provides information about the number of OTUs shared between
methods. This function allows to identify the OTUs that are specific to
each method and also the ones that are not shared among any method.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Exclusion plot </span></span>
<span><span class="fu"><a href="../reference/exclusion_plt.html">exclusion_plt</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span></span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;"></p>
<p>Besides to the previously mentioned functions, <code>dar</code> also
includes the function <code><a href="../reference/corr_heatmap.html">corr_heatmap()</a></code>, which allows for
visualization of the overlap of significant OTUs between tested methods.
This function can provide similar information to the previous plots, but
in some cases it may be easier to interpret. comprehensive view of the
results.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Correlation heatmap</span></span>
<span><span class="va">corr_heat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corr_heatmap.html">corr_heatmap</a></span><span class="op">(</span><span class="va">da_results</span>, font_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> </span>
<span><span class="va">corr_heat</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:768px;height:576px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":[{"x":[1.5,1,null,1,1,null,1.5,2,null,2,2],"y":[0.43422331915117574,0.43422331915117574,null,0.43422331915117574,0,null,0.43422331915117574,0.43422331915117574,null,0.43422331915117574,0],"text":"y: 0.434","type":"scatter","mode":"lines","line":{"width":2.2677165354330708,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.5,1,2],"y":[0.43422331915117574,0,0],"text":["y: 0.434","y: 0.000","y: 0.000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.8897637795275593,"color":"transparent"}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"type":"scatter","mode":"markers","marker":{"color":"rgba(44,160,44,1)","line":{"color":"rgba(44,160,44,1)"}},"error_y":{"color":"rgba(44,160,44,1)"},"error_x":{"color":"rgba(44,160,44,1)"},"line":{"color":"rgba(44,160,44,1)"},"xaxis":"x2","yaxis":"y","frame":null},{"x":[1,1,2,2],"y":[1,2,1,2],"text":["row: metagenomeseq__Spina_santa<br>column: metagenomeseq__Spina_santa<br>value: 1.000 <br> Point size :  1.000","row: maaslin__Dutch_Baby_Pancake<br>column: metagenomeseq__Spina_santa<br>value: 0.693 <br> Point size :  0.693","row: metagenomeseq__Spina_santa<br>column: maaslin__Dutch_Baby_Pancake<br>value: 0.693 <br> Point size :  0.693","row: maaslin__Dutch_Baby_Pancake<br>column: maaslin__Dutch_Baby_Pancake<br>value: 1.000 <br> Point size :  1.000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":["rgba(165,0,38,1)","rgba(253,178,100,1)","rgba(253,178,100,1)","rgba(165,0,38,1)"],"opacity":1,"size":[22.677165354330711,3.7795275590551185,3.7795275590551185,22.677165354330711],"symbol":"circle","line":{"width":1.8897637795275593,"color":["rgba(165,0,38,1)","rgba(253,178,100,1)","rgba(253,178,100,1)","rgba(165,0,38,1)"]}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1],"y":[1],"name":"9fd1a829bc79f58faf1a621f1a26f9fc","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#006837"],[0.0033444816053511705,"#016A38"],[0.006688963210702341,"#016B39"],[0.010033444816053512,"#026D39"],[0.013377926421404682,"#036E3A"],[0.016722408026755852,"#04703B"],[0.020066889632107024,"#04713C"],[0.023411371237458192,"#05733D"],[0.026755852842809364,"#06753E"],[0.030100334448160536,"#07763E"],[0.033444816053511704,"#08783F"],[0.036789297658862873,"#097940"],[0.040133779264214048,"#0A7B41"],[0.043478260869565216,"#0B7C42"],[0.046822742474916385,"#0C7E43"],[0.05016722408026756,"#0D8043"],[0.053511705685618728,"#0E8144"],[0.056856187290969896,"#0F8345"],[0.060200668896321072,"#108546"],[0.06354515050167224,"#118647"],[0.066889632107023408,"#128848"],[0.070234113712374577,"#128948"],[0.073578595317725745,"#138B49"],[0.076923076923076927,"#148D4A"],[0.080267558528428096,"#158E4B"],[0.083612040133779264,"#16904C"],[0.086956521739130432,"#17924D"],[0.090301003344481601,"#18934E"],[0.093645484949832769,"#18954E"],[0.096989966555183951,"#19974F"],[0.10033444816053512,"#1A9850"],[0.10367892976588629,"#1F9951"],[0.10702341137123746,"#229B51"],[0.11036789297658862,"#269C52"],[0.11371237458193979,"#299D53"],[0.11705685618729096,"#2C9E53"],[0.12040133779264214,"#2F9F54"],[0.12374581939799331,"#32A154"],[0.12709030100334448,"#35A255"],[0.13043478260869565,"#38A356"],[0.13377926421404682,"#3AA456"],[0.13712374581939799,"#3DA657"],[0.14046822742474915,"#3FA758"],[0.14381270903010032,"#42A858"],[0.14715719063545149,"#44A959"],[0.15050167224080269,"#46AB5A"],[0.15384615384615385,"#49AC5A"],[0.15719063545150502,"#4BAD5B"],[0.16053511705685619,"#4DAE5B"],[0.16387959866220736,"#4FB05C"],[0.16722408026755853,"#52B15D"],[0.1705685618729097,"#54B25D"],[0.17391304347826086,"#56B35E"],[0.17725752508361203,"#58B55F"],[0.1806020066889632,"#5AB65F"],[0.18394648829431437,"#5CB760"],[0.18729096989966554,"#5EB861"],[0.19063545150501671,"#60B961"],[0.1939799331103679,"#62BB62"],[0.19732441471571907,"#64BC62"],[0.20066889632107024,"#66BD63"],[0.20401337792642141,"#69BE63"],[0.20735785953177258,"#6BBF64"],[0.21070234113712374,"#6DC064"],[0.21404682274247491,"#70C164"],[0.21739130434782608,"#72C264"],[0.22073578595317725,"#74C364"],[0.22408026755852842,"#76C465"],[0.22742474916387959,"#79C565"],[0.23076923076923075,"#7BC665"],[0.23411371237458192,"#7DC765"],[0.23745819397993312,"#7FC766"],[0.24080267558528429,"#81C866"],[0.24414715719063546,"#83C966"],[0.24749163879598662,"#86CA66"],[0.25083612040133779,"#88CB67"],[0.25418060200668896,"#8ACC67"],[0.25752508361204013,"#8CCD67"],[0.2608695652173913,"#8ECE67"],[0.26421404682274247,"#90CF68"],[0.26755852842809363,"#92D068"],[0.2709030100334448,"#94D168"],[0.27424749163879597,"#96D268"],[0.27759197324414714,"#98D368"],[0.28093645484949831,"#9AD469"],[0.28428093645484948,"#9CD569"],[0.28762541806020064,"#9FD669"],[0.29096989966555181,"#A1D669"],[0.29431438127090298,"#A3D76A"],[0.29765886287625415,"#A5D86A"],[0.30100334448160537,"#A7D96A"],[0.30434782608695654,"#A8DA6B"],[0.30769230769230771,"#AADB6D"],[0.31103678929765888,"#ACDB6E"],[0.31438127090301005,"#ADDC6F"],[0.31772575250836121,"#AFDD70"],[0.32107023411371238,"#B1DE71"],[0.32441471571906355,"#B3DE72"],[0.32775919732441472,"#B4DF73"],[0.33110367892976589,"#B6E074"],[0.33444816053511706,"#B8E175"],[0.33779264214046822,"#B9E176"],[0.34113712374581939,"#BBE278"],[0.34448160535117056,"#BDE379"],[0.34782608695652173,"#BFE47A"],[0.3511705685618729,"#C0E47B"],[0.35451505016722407,"#C2E57C"],[0.35785953177257523,"#C4E67D"],[0.3612040133779264,"#C5E77E"],[0.36454849498327757,"#C7E77F"],[0.36789297658862874,"#C9E880"],[0.37123745819397991,"#CAE981"],[0.37458193979933108,"#CCE983"],[0.37792642140468224,"#CEEA84"],[0.38127090301003341,"#D0EB85"],[0.38461538461538458,"#D1EC86"],[0.38795986622073581,"#D3EC87"],[0.39130434782608697,"#D5ED88"],[0.39464882943143814,"#D6EE89"],[0.39799331103678931,"#D8EF8A"],[0.40133779264214048,"#DAEF8C"],[0.40468227424749165,"#DBF08D"],[0.40802675585284282,"#DCF08F"],[0.41137123745819398,"#DDF191"],[0.41471571906354515,"#DFF193"],[0.41806020066889632,"#E0F294"],[0.42140468227424749,"#E1F296"],[0.42474916387959866,"#E3F398"],[0.42809364548494983,"#E4F49A"],[0.43143812709030099,"#E5F49B"],[0.43478260869565216,"#E6F59D"],[0.43812709030100333,"#E8F59F"],[0.4414715719063545,"#E9F6A1"],[0.44481605351170567,"#EAF6A2"],[0.44816053511705684,"#EBF7A4"],[0.451505016722408,"#EDF7A6"],[0.45484949832775917,"#EEF8A8"],[0.45819397993311034,"#EFF8A9"],[0.46153846153846151,"#F1F9AB"],[0.46488294314381268,"#F2F9AD"],[0.46822742474916385,"#F3FAAF"],[0.47157190635451507,"#F4FAB0"],[0.47491638795986624,"#F6FBB2"],[0.47826086956521741,"#F7FCB4"],[0.48160535117056857,"#F8FCB5"],[0.48494983277591974,"#F9FDB7"],[0.48829431438127091,"#FBFDB9"],[0.49163879598662208,"#FCFEBB"],[0.49498327759197325,"#FDFEBC"],[0.49832775919732442,"#FEFFBE"],[0.50167224080267558,"#FFFEBE"],[0.50501672240802675,"#FFFDBC"],[0.50836120401337792,"#FFFCBB"],[0.51170568561872909,"#FFFBB9"],[0.51505016722408026,"#FFFAB7"],[0.51839464882943143,"#FFF9B5"],[0.52173913043478259,"#FFF8B4"],[0.52508361204013376,"#FFF7B2"],[0.52842809364548493,"#FFF6B0"],[0.5317725752508361,"#FFF5AE"],[0.53511705685618727,"#FFF4AD"],[0.53846153846153844,"#FFF3AB"],[0.5418060200668896,"#FFF2A9"],[0.54515050167224077,"#FFF1A7"],[0.54849498327759194,"#FFF0A6"],[0.55183946488294311,"#FFEFA4"],[0.55518394648829428,"#FFEEA2"],[0.55852842809364545,"#FFEDA1"],[0.56187290969899661,"#FFEC9F"],[0.56521739130434778,"#FFEB9D"],[0.56856187290969895,"#FFEA9B"],[0.57190635451505012,"#FFE99A"],[0.57525083612040129,"#FFE898"],[0.57859531772575246,"#FFE796"],[0.58193979933110362,"#FFE694"],[0.58528428093645479,"#FEE593"],[0.58862876254180596,"#FEE391"],[0.59197324414715713,"#FEE28F"],[0.5953177257525083,"#FEE18D"],[0.59866220735785947,"#FEE08C"],[0.60200668896321075,"#FEDF8A"],[0.60535117056856191,"#FEDD89"],[0.60869565217391308,"#FEDC87"],[0.61204013377926425,"#FEDA86"],[0.61538461538461542,"#FED884"],[0.61872909698996659,"#FED783"],[0.62207357859531776,"#FED582"],[0.62541806020066892,"#FED380"],[0.62876254180602009,"#FED27F"],[0.63210702341137126,"#FED07D"],[0.63545150501672243,"#FECE7C"],[0.6387959866220736,"#FECD7B"],[0.64214046822742477,"#FECB79"],[0.64548494983277593,"#FEC978"],[0.6488294314381271,"#FEC876"],[0.65217391304347827,"#FEC675"],[0.65551839464882944,"#FEC473"],[0.65886287625418061,"#FEC372"],[0.66220735785953178,"#FEC171"],[0.66555183946488294,"#FEBF6F"],[0.66889632107023411,"#FEBE6E"],[0.67224080267558528,"#FEBC6C"],[0.67558528428093645,"#FEBA6B"],[0.67892976588628762,"#FEB96A"],[0.68227424749163879,"#FEB768"],[0.68561872909698995,"#FEB567"],[0.68896321070234112,"#FDB466"],[0.69230769230769229,"#FDB264"],[0.69565217391304346,"#FDB063"],[0.69899665551839463,"#FDAF61"],[0.7023411371237458,"#FDAD60"],[0.70568561872909696,"#FDAB5F"],[0.70903010033444813,"#FCA85E"],[0.7123745819397993,"#FCA65D"],[0.71571906354515047,"#FCA45C"],[0.71906354515050164,"#FCA25B"],[0.72240802675585281,"#FCA05A"],[0.72575250836120397,"#FB9E59"],[0.72909698996655514,"#FB9C58"],[0.73244147157190631,"#FB9A57"],[0.73578595317725748,"#FB9856"],[0.73913043478260865,"#FA9655"],[0.74247491638795982,"#FA9354"],[0.74581939799331098,"#FA9153"],[0.74916387959866215,"#F98F52"],[0.75250836120401332,"#F98D51"],[0.75585284280936449,"#F98B50"],[0.75919732441471566,"#F8894F"],[0.76254180602006683,"#F8874E"],[0.76588628762541799,"#F8844D"],[0.76923076923076916,"#F7824C"],[0.77257525083612044,"#F7804B"],[0.77591973244147161,"#F77E4A"],[0.77926421404682278,"#F67B49"],[0.78260869565217395,"#F67948"],[0.78595317725752512,"#F67747"],[0.78929765886287628,"#F57546"],[0.79264214046822745,"#F57245"],[0.79598662207357862,"#F47044"],[0.79933110367892979,"#F46D43"],[0.80267558528428096,"#F36C42"],[0.80602006688963213,"#F26A41"],[0.80936454849498329,"#F16840"],[0.81270903010033446,"#F0663F"],[0.81605351170568563,"#EF643E"],[0.8193979933110368,"#EF633D"],[0.82274247491638797,"#EE613C"],[0.82608695652173914,"#ED5F3B"],[0.8294314381270903,"#EC5D3A"],[0.83277591973244147,"#EB5B3A"],[0.83612040133779264,"#EA5939"],[0.83946488294314381,"#E95738"],[0.84280936454849498,"#E85537"],[0.84615384615384615,"#E75336"],[0.84949832775919731,"#E65135"],[0.85284280936454848,"#E54F34"],[0.85618729096989965,"#E44D33"],[0.85953177257525082,"#E34B32"],[0.86287625418060199,"#E24931"],[0.86622073578595316,"#E14730"],[0.86956521739130432,"#E0452F"],[0.87290969899665549,"#DF432E"],[0.87625418060200666,"#DE412D"],[0.87959866220735783,"#DD3F2C"],[0.882943143812709,"#DC3C2C"],[0.88628762541806017,"#DB3A2B"],[0.88963210702341133,"#DA382A"],[0.8929765886287625,"#D93529"],[0.89632107023411367,"#D83328"],[0.89966555183946484,"#D73027"],[0.90301003344481601,"#D52F27"],[0.90635451505016718,"#D42E27"],[0.90969899665551834,"#D22C27"],[0.91304347826086951,"#D02B27"],[0.91638795986622068,"#CF2A27"],[0.91973244147157185,"#CD2927"],[0.92307692307692302,"#CB2727"],[0.92642140468227419,"#CA2627"],[0.92976588628762535,"#C82527"],[0.93311036789297652,"#C62427"],[0.93645484949832769,"#C52227"],[0.93979933110367886,"#C32127"],[0.94314381270903014,"#C11F27"],[0.94648829431438131,"#BF1E27"],[0.94983277591973247,"#BE1D27"],[0.95317725752508364,"#BC1B27"],[0.95652173913043481,"#BA1A27"],[0.95986622073578598,"#B91827"],[0.96321070234113715,"#B71627"],[0.96655518394648832,"#B51527"],[0.96989966555183948,"#B41327"],[0.97324414715719065,"#B21127"],[0.97658862876254182,"#B10F27"],[0.97993311036789299,"#AF0D26"],[0.98327759197324416,"#AD0B26"],[0.98662207357859533,"#AC0926"],[0.98996655518394649,"#AA0726"],[0.99331103678929766,"#A80426"],[0.99665551839464883,"#A70226"],[1,"#A50026"]],"colorbar":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"thickness":23.040000000000003,"title":"value","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"tickmode":"array","ticktext":["0.00","0.25","0.50","0.75","1.00"],"tickvals":[0.0016666666666666668,0.2508333333333333,0.49999999999999994,0.74916666666666665,0.99833333333333329],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y2","frame":null},{"x":[0.43422331915117574,0.43422331915117574,null,0.43422331915117574,0,null,0.43422331915117574,0.43422331915117574,null,0.43422331915117574,0],"y":[1.5,1,null,1,1,null,1.5,2,null,2,2],"text":"y: 0.434","type":"scatter","mode":"lines","line":{"width":2.2677165354330708,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.43422331915117574,0,0],"y":[1.5,1,2],"text":["y: 0.434","y: 0.000","y: 0.000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.8897637795275593,"color":"transparent"}},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"xaxis":{"domain":[0,0.80000000000000004],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,2.6000000000000001],"tickmode":"array","ticktext":["metagenomeseq__Spina_santa","maaslin__Dutch_Baby_Pancake"],"tickvals":[1,2],"categoryorder":"array","categoryarray":["metagenomeseq__Spina_santa","maaslin__Dutch_Baby_Pancake"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"Arial","size":13.283520132835205},"tickangle":-45,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"xaxis2":{"domain":[0.80000000000000004,1],"automargin":true,"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"y2","type":"linear","autorange":false,"range":[0,0.43422331915117574],"tickmode":"array","ticktext":["0.0","0.1","0.2","0.3","0.4"],"tickvals":[0,0.10000000000000001,0.20000000000000001,0.30000000000000004,0.40000000000000002],"categoryorder":"array","categoryarray":["0.0","0.1","0.2","0.3","0.4"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":{"text":"","font":{"color":null,"family":null,"size":0}},"hoverformat":".2f"},"yaxis2":{"domain":[0,0.80000000000000004],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,2.6000000000000001],"tickmode":"array","ticktext":["metagenomeseq__Spina_santa","maaslin__Dutch_Baby_Pancake"],"tickvals":[1,2],"categoryorder":"array","categoryarray":["metagenomeseq__Spina_santa","maaslin__Dutch_Baby_Pancake"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"Arial","size":13.283520132835205},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"yaxis":{"domain":[0.80000000000000004,1],"automargin":true,"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"x","type":"linear","autorange":false,"range":[0,0.43422331915117574],"tickmode":"array","ticktext":["0.0","0.1","0.2","0.3","0.4"],"tickvals":[0,0.10000000000000001,0.20000000000000001,0.30000000000000004,0.40000000000000002],"categoryorder":"array","categoryarray":["0.0","0.1","0.2","0.3","0.4"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":{"text":"","font":{"color":null,"family":null,"size":0}},"hoverformat":".2f"},"annotations":[],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.80000000000000004,"y0":0.80000000000000004,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.80000000000000004,"y0":0,"y1":0.80000000000000004},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.80000000000000004,"x1":1,"y0":0,"y1":0.80000000000000004}],"images":[],"margin":{"t":0,"r":null,"b":215.19302615193033,"l":215.19302615193033},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"Point size<br />value","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"y":1,"yanchor":"top"},"hovermode":"closest","barmode":"relative","paper_bgcolor":"rgba(255,255,255,1)","title":""},"attrs":{"61444a4acc08":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"scatter"},"614466cf6c22":{"colour":{},"shape":{},"size":{},"x":{},"y":{}},"61445bd010":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20]},"614416b36939":{"x":{},"y":{},"colour":{},"text":{},"size":{},"type":"scatter"},"6144259748b6":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"scatter"},"6144514cbcba":{"colour":{},"shape":{},"size":{},"x":{},"y":{}}},"source":"A","config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false,"displaylogo":false,"modeBarButtonsToRemove":["sendDataToCloud","select2d","lasso2d","autoScale2d","hoverClosestCartesian","hoverCompareCartesian","sendDataToCloud"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"subplot":true,"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p>Finally, <code>dar</code> also includes the function
<code><a href="../reference/mutual_plt.html">mutual_plt()</a></code>, which plots the number of differential
abundant features mutually found by a defined number of methods, colored
by the differential abundance direction and separated by comparison. The
resulting graph allows us to see that the features detected correspond
mainly to the comparisons between hts vs msm and msm vs pwid.
Additionally, the graph also allows us to observe the direction of the
effect; whether a specific OTU is enriched or depleted for each
comparison.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Mutual plot</span></span>
<span><span class="fu"><a href="../reference/mutual_plt.html">mutual_plt</a></span><span class="op">(</span></span>
<span>  <span class="va">da_results</span>, </span>
<span>  count_cutoff <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/steps_ids.html">steps_ids</a></span><span class="op">(</span><span class="va">da_results</span>, type <span class="op">=</span> <span class="st">"da"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="define-a-consesus-strategy-using-bake">Define a consesus strategy using bake<a class="anchor" aria-label="anchor" href="#define-a-consesus-strategy-using-bake"></a>
</h2>
<p>After visually inspecting the results from running all the
differential analysis methods on our data, we have the necessary
information to define a consensus strategy that fits our dataset. In our
case, we will retain all the methods. However if one or more methods are
not desired, the <code><a href="../reference/bake.html">bake()</a></code> function includes the
<code>exclude</code> parameter, which allows to exclude specific
methods.</p>
<p>Additionally, the <code><a href="../reference/bake.html">bake()</a></code> function allows to further
refine the consensus strategy through its parameters, such as
<code>count_cutoff</code>, which indicates the minimum number of methods
in which an OTU must be present, and <code>weights</code>, a named
vector with the ponderation value for each method. However, for
simplicity, these parameters are not used in this example.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define consensus strategy</span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bake.html">bake</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Results</span> ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">355</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> metagenomeseq__Spina_santa <span style="color: #555555;">diff_taxa = 294</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> maaslin__Dutch_Baby_Pancake <span style="color: #555555;">diff_taxa = 235</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> 210 taxa are present in all tested methods </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Bakes:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> <span style="color: #0000BB; font-weight: bold;">1 -&gt;</span> <span style="color: #555555;">count_cutoff: NULL, weights: NULL, exclude: NULL, id: bake__Cornish_pasty</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="extract-results">Extract results<a class="anchor" aria-label="anchor" href="#extract-results"></a>
</h2>
<p>To conclude, we can extract the final results using the
<code><a href="../reference/cool.html">cool()</a></code> function. This function takes a
<code>PrepRecipe</code> object and the ID of the bake to be used as
input (by default it is 1, but if you have multiple consensus
strategies, you can change it to extract the desired results).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract results for bake id 1</span></span>
<span><span class="va">f_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cool.html">cool</a></span><span class="op">(</span><span class="va">da_results</span>, bake <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">f_results</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 210 × 2</span></span></span>
<span><span class="co">#&gt;    taxa_id taxa                   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Otu_70  Bacteroides_sp_CAG_598 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Otu_73  Bacteroides_sp_D2      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Otu_369 Dialister_sp_CAG_357   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Otu_121 Alistipes_sp_An31A     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Otu_63  Bacteroides_plebeius   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Otu_216 Clostridium_sp_CAG_632 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Otu_257 Butyrivibrio_sp_CAG_318</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Otu_137 Enterococcus_avium     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Otu_49  Bacteroides_coprocola  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Otu_441 Brachyspira_sp_CAG_700 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 200 more rows</span></span></span></code></pre></div>
<p>To further visualize the results, the <code><a href="../reference/abundance_plt.html">abundance_plt()</a></code>
function can be utilized to visualize the differences in abundance of
the differential abundant taxa.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Ids for Bacteroide and Provotella species </span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">f_results</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">taxa</span>, <span class="st">"Bacteroi.*|Prevote.*"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">taxa_id</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## Abundance plot as boxplot</span></span>
<span><span class="fu"><a href="../reference/abundance_plt.html">abundance_plt</a></span><span class="op">(</span><span class="va">da_results</span>, taxa_ids <span class="op">=</span> <span class="va">ids</span>, type <span class="op">=</span> <span class="st">"boxplot"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-13-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.0 RC (2024-04-16 r86441)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-04-24</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  ade4               1.7-22     <span style="color: #555555;">2023-02-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ape                5.8        <span style="color: #555555;">2024-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  assertthat         0.2.1      <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase            2.63.1     <span style="color: #555555;">2024-03-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics       0.49.1     <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  biomformat         1.31.0     <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings         2.71.5     <span style="color: #555555;">2024-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  bitops             1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  brio               1.1.4      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib              0.7.0      <span style="color: #555555;">2024-03-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ca                 0.71.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem             1.0.8      <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                3.6.2      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cluster            2.1.6      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  codetools          0.2-20     <span style="color: #555555;">2024-03-31</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  colorspace         2.1-0      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crayon             1.5.2      <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crosstalk          1.2.1      <span style="color: #555555;">2023-11-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dar              * 0.99.13    <span style="color: #555555;">2024-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  data.table         1.15.4     <span style="color: #555555;">2024-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  dendextend         1.17.1     <span style="color: #555555;">2023-03-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  desc               1.4.3      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  devtools           2.4.5      <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  digest             0.6.35     <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dplyr              1.1.4      <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ellipsis           0.3.2      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  evaluate           0.23       <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fansi              1.0.6      <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  farver             2.1.1      <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fastmap            1.1.1      <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  foreach            1.5.2      <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                 1.6.3      <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  furrr              0.3.1      <span style="color: #555555;">2022-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  future             1.33.2     <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  generics           0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb       1.39.14    <span style="color: #555555;">2024-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData   1.2.12     <span style="color: #555555;">2024-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggplot2            3.5.1      <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  globals            0.16.3     <span style="color: #555555;">2024-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  glue               1.7.0      <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gridExtra          2.3        <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gtable             0.3.5      <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  heatmaply          1.5.0      <span style="color: #555555;">2023-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  highr              0.10       <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmltools          0.5.8.1    <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets        1.6.4      <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httpuv             1.6.15     <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr               1.4.7      <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  igraph             2.0.3      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  IRanges            2.37.1     <span style="color: #555555;">2024-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  iterators          1.0.14     <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib          0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite           1.8.8      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr              1.46       <span style="color: #555555;">2024-04-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  labeling           0.4.3      <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  later              1.3.2      <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice            0.22-6     <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lazyeval           0.2.2      <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle          1.0.4      <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  listenv            0.9.1      <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magrittr           2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  MASS               7.3-60.2   <span style="color: #555555;">2024-04-18</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">#&gt;  Matrix             1.7-0      <span style="color: #555555;">2024-03-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  memoise            2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  mgcv               1.9-1      <span style="color: #555555;">2023-12-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  microbiome         1.25.0     <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  mime               0.12       <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  miniUI             0.1.1.1    <span style="color: #555555;">2018-05-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  multtest           2.59.0     <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  munsell            0.5.1      <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  nlme               3.1-164    <span style="color: #555555;">2023-11-27</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  parallelly         1.37.1     <span style="color: #555555;">2024-02-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  permute            0.9-7      <span style="color: #555555;">2022-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  phyloseq           1.47.0     <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  pillar             1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgbuild           1.4.4      <span style="color: #555555;">2024-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig          2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown            <span style="color: #BB00BB; font-weight: bold;">2.0.9.9000</span> <span style="color: #555555;">2024-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/pkgdown@4e7ffec)</span></span></span>
<span><span class="co">#&gt;  pkgload            1.3.4      <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plotly             4.10.4     <span style="color: #555555;">2024-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plyr               1.8.9      <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  profvis            0.3.8      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  promises           1.3.0      <span style="color: #555555;">2024-04-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  purrr              1.0.2      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                 2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg               1.3.0      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer       1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp               1.0.12     <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RCurl              1.98-1.14  <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  registry           0.5-1      <span style="color: #555555;">2019-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  remotes            2.5.0      <span style="color: #555555;">2024-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  reshape2           1.4.4      <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5              2.47.7     <span style="color: #555555;">2024-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5filters       1.15.5     <span style="color: #555555;">2024-04-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib           1.25.3     <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  rlang              1.1.3      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown          2.26       <span style="color: #555555;">2024-03-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rtsne              0.17       <span style="color: #555555;">2023-12-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  S4Vectors          0.41.6     <span style="color: #555555;">2024-04-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  sass               0.4.9      <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scales             1.3.0      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  seriation          1.5.5      <span style="color: #555555;">2024-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo        1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shiny              1.8.1.1    <span style="color: #555555;">2024-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringi            1.8.3      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringr            1.5.1      <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  survival           3.5-8      <span style="color: #555555;">2024-02-14</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  systemfonts        1.0.6      <span style="color: #555555;">2024-03-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  testthat           3.2.1.1    <span style="color: #555555;">2024-04-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping        0.3.7      <span style="color: #555555;">2023-10-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tibble             3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyr              1.3.1      <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect         1.2.1      <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.5.0)</span></span></span>
<span><span class="co">#&gt;  TSP                1.2-4      <span style="color: #555555;">2023-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils         0.99.7     <span style="color: #555555;">2024-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UpSetR             1.4.0      <span style="color: #555555;">2019-05-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  urlchecker         1.0.1      <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  usethis            2.2.3      <span style="color: #555555;">2024-02-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  utf8               1.2.4      <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vctrs              0.6.5      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vegan              2.6-4      <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridis            0.6.5      <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite        0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  webshot            0.5.5      <span style="color: #555555;">2023-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  withr              3.0.0      <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xfun               0.43       <span style="color: #555555;">2024-03-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xtable             1.8-4      <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector            0.43.1     <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  yaml               2.3.8      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc           1.49.3     <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.5.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Francesc Catala-Moll.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
