<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="dar">
<title>Workflow with real data • dar</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Workflow with real data">
<meta property="og:description" content="dar">
<meta property="og:image" content="https://microbialgenomics-irsicaixaorg.github.io/dar/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dar</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/dar.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/article.html">Workflow with real data</a>
    <a class="dropdown-item" href="../articles/data_import.html">Data Import</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="article_files/htmlwidgets-1.6.1/htmlwidgets.js"></script><script src="article_files/plotly-binding-4.10.1/plotly.js"></script><script src="article_files/typedarray-0.1/typedarray.min.js"></script><link href="article_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="article_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><link href="article_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="article_files/plotly-main-2.11.1/plotly-latest.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Workflow with real data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/blob/HEAD/vignettes/article.Rmd" class="external-link"><code>vignettes/article.Rmd</code></a></small>
      <div class="d-none name"><code>article.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<p>To illustrate the functionality of the ‘dar’ package, this study will
use the data set from (Noguera-Julian, M., et al. 2016). The authors of
this study found that men who have sex with men (MSM) predominantly
belonged to the Prevotella-rich enterotype whereas most non-MSM subjects
were enriched in Bacteroides, independently of HIV-1 status. This result
highlights the potential impact of sexual orientation on the gut
microbiome and emphasizes the importance of controlling for such
variables in microbiome research. Using the ‘dar’ package, we will
conduct a differential abundance analysis to further explore this
finding and uncover potential microbial biomarkers associated with this
specific population.</p>
<div class="section level2">
<h2 id="load-dar-package-and-data">Load dar package and data<a class="anchor" aria-label="anchor" href="#load-dar-package-and-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar" class="external-link">dar</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://plotly-r.com" class="external-link">plotly</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metaHIV_phy"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaHIV_phy</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 451 taxa and 156 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 156 samples by 3 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 451 taxa by 7 taxonomic ranks ]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="recipe-initialization">Recipe initialization<a class="anchor" aria-label="anchor" href="#recipe-initialization"></a>
</h2>
<p>To begin the analysis process with the ‘dar’ package, the first step
is to initialize a recipe object, which is an S4 class. This recipe
object serves as a blueprint for the data preparation steps required for
the differential abundan ce analysis. The initialization of the recipe
object is done through the function recipe(), which takes as inputs a
phyloseq object, the name of the categorical variable of interest and
the taxonomic level at which the differential abundance analyses are to
be performed. As previously mentioned, we will use the data set from
(Noguera-Julian, M., et al. 2016) and the variable of interest
“RiskGroup2” containing the categories: men who have sex with men (msm),
non-MSM (hts) and people who inject drugs (pwid) and we will perform the
analysis at the species level.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Recipe initialization</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/recipe.html">recipe</a></span><span class="op">(</span><span class="va">metaHIV_phy</span>, var_info <span class="op">=</span> <span class="st">"RiskGroup2"</span>, tax_info <span class="op">=</span> <span class="st">"Species"</span><span class="op">)</span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="recipe-qc-and-preprocessing-steps-definition">Recipe QC and preprocessing steps definition<a class="anchor" aria-label="anchor" href="#recipe-qc-and-preprocessing-steps-definition"></a>
</h2>
<p>Once the recipe object has been initialized, the next step is to
populate it with steps. Steps are the methods that will be applied to
the data stored in the recipe. There are two types of steps:
preprocessing (prepro) and differential abundance (da) steps. Initially,
we will focus on the prepro steps which are used to modify the data
loaded into the recipe, which will then be used for the da steps. The
‘dar’ package includes 3 main preprocessing functionalities:
step_subset_taxa, which is used for subsetting columns and values in the
taxon table connected to the phyloseq object, step_filter_taxa, which is
used to filter the OTUs, and step_rarefaction, which is used to resample
the OTU table to ensure that all samples have the same library size.
These functionalities allow for a high level of flexibility and
customization in the data preparation process before performing the
differential abundance analysis.</p>
<p>In addition to the preprocessing steps, the ‘dar’ package also
incorporates the function phy_qc() which returns a table with a set of
metrics that allow for informed decisions to be made about the data
preprocessing that will be done. In our case, we decided to use the
step_subset_taxa function to retain only those observations annotated
within the realm of bacteria and archaea. We also used the
step_filter_taxa function to retain only those OTUs with at least 1% of
the samples with values greater than 0. This approach ensured that we
were working with a high-quality, informative subset of the data, which
improved the overall accuracy and reliability of the differential
abundance analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## QC </span></span>
<span><span class="fu"><a href="../reference/phy_qc.html">phy_qc</a></span><span class="op">(</span><span class="va">rec</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 10</span></span></span>
<span><span class="co">#&gt;   var_lev…¹     n n_zero pct_z…² pct_a…³ pct_s…⁴ pct_d…⁵ count…⁶ count…⁷ count…⁸</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> all       <span style="text-decoration: underline;">70</span>356  <span style="text-decoration: underline;">57</span>632    81.9     0      20.6    8.87  <span style="text-decoration: underline;">23</span>805.       0  3.34<span style="color: #949494;">e</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> hts       <span style="text-decoration: underline;">18</span>491  <span style="text-decoration: underline;">15</span>108    81.7    24.2    22.8    8.43  <span style="text-decoration: underline;">36</span>017.       0  4.52<span style="color: #949494;">e</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> msm       <span style="text-decoration: underline;">45</span>100  <span style="text-decoration: underline;">37</span>019    82.1    16.0    20.2    9.53  <span style="text-decoration: underline;">17</span>104.       0  2.74<span style="color: #949494;">e</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> pwid       <span style="text-decoration: underline;">6</span>765   <span style="text-decoration: underline;">5</span>505    81.4    41.2    16.6    9.31  <span style="text-decoration: underline;">35</span>097.       0  4.13<span style="color: #949494;">e</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable names ¹​var_levels, ²​pct_zero, ³​pct_all_zero,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁴​pct_singletons, ⁵​pct_doubletons, ⁶​count_mean, ⁷​count_min, ⁸​count_max</span></span></span>
<span></span>
<span><span class="co">## Adding prepro steps</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">rec</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_subset_taxa.html">step_subset_taxa</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="st">'Kingdom %in% c("Bacteria", "Archaea")'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_filter_taxa.html">step_filter_taxa</a></span><span class="op">(</span>.f <span class="op">=</span> <span class="st">"function(x) sum(x &gt; 0) &gt;= (0.1 * length(x))"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Cherry_pie</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_filter_taxa() <span style="color: #555555;">id = filter_taxa__Briouat</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-differential-analysis-da-steps">Define Differential Analysis (DA) steps<a class="anchor" aria-label="anchor" href="#define-differential-analysis-da-steps"></a>
</h2>
<p>Once data is preprocessed and cleaned, the next step is to add the da
steps. The ‘dar’ package incorporates multiple methods to analyze the
data, including: ALDEx2, ANCOM-BC, corncob, DESeq2, Lefse, MAaslin2,
MetagenomeSeq, and Wilcox. These methods provide a range of options for
uncovering potential microbial biomarkers associated with the variable
of interest. To ensure consistency across methods, we decided not to use
default parameters, but to set the min_prevalence parameter to 0 for
MAaslin2, and the rm_zeros parameter to 0.01 for MetagenomeSeq, since it
was observed that the pct_all_zeros value was not equal to 0 in some
levels of the categorical variable in the results of phy_qc(). This
approach ensured that the analysis was consistent across all methods and
that the results were interpretable.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## DA steps definition</span></span>
<span><span class="va">rec</span> <span class="op">&lt;-</span> <span class="va">rec</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_ancom.html">step_ancom</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/step_aldex.html">step_aldex</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/step_deseq.html">step_deseq</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/step_corncob.html">step_corncob</a></span><span class="op">(</span>filter_discriminant <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/step_metagenomeseq.html">step_metagenomeseq</a></span><span class="op">(</span>rm_zeros <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/step_maaslin.html">step_maaslin</a></span><span class="op">(</span>min_prevalence <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/step_lefse.html">step_lefse</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_wilcox.html">step_wilcox</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rec</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Cherry_pie</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_filter_taxa() <span style="color: #555555;">id = filter_taxa__Briouat</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_ancom() <span style="color: #555555;">id = ancom__Roti_john</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_aldex() <span style="color: #555555;">id = aldex__Eccles_cake</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_deseq() <span style="color: #555555;">id = deseq__Öçpoçmaq</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_corncob() <span style="color: #555555;">id = corncob__Paste</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_metagenomeseq() <span style="color: #555555;">id = metagenomeseq__Pain_aux_raisins</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_maaslin() <span style="color: #555555;">id = maaslin__Nazook</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_lefse() <span style="color: #555555;">id = lefse__Filo</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_wilcox() <span style="color: #555555;">id = wilcox__Cannoli_siciliani</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prep-recipe">Prep recipe<a class="anchor" aria-label="anchor" href="#prep-recipe"></a>
</h2>
<p>Once the recipe has been defined, the next step is to execute all the
steps defined in the recipe. This is done through the function prep().
Internally, it first executes the preprocessing steps, which modify the
phyloseq object stored in the recipe. Then, using the modified phyloseq,
it executes each of the defined differential abundance methods. To speed
up the execution time, the prep() function includes the option to run in
parallel. The resulting object has class ‘prep_recipe’ and when printed
in the terminal, it displays the number of taxa detected as significant
in each of the methods and also the total number of taxa shared across
all methods. This allows for a provisional overview of the results and a
comparison between methods.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Execute in parallel</span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep.html">prep</a></span><span class="op">(</span><span class="va">rec</span>, parallel <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: There was 1 warning in `dplyr::mutate()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> In argument: `dplyr::across(...)`.</span></span>
<span><span class="co">#&gt; Caused by warning:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Using `across()` without supplying `.cols` was deprecated in dplyr 1.1.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please supply `.cols` instead.</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Results</span> ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">174</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> ancom__Roti_john <span style="color: #555555;">diff_taxa = 63</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> aldex__Eccles_cake <span style="color: #555555;">diff_taxa = 112</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> deseq__Öçpoçmaq <span style="color: #555555;">diff_taxa = 92</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> corncob__Paste <span style="color: #555555;">diff_taxa = 125</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> metagenomeseq__Pain_aux_raisins <span style="color: #555555;">diff_taxa = 136</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> maaslin__Nazook <span style="color: #555555;">diff_taxa = 148</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> lefse__Filo <span style="color: #555555;">diff_taxa = 121</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> wilcox__Cannoli_siciliani <span style="color: #555555;">diff_taxa = 123</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> 26 taxa are present in all tested methods</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="default-results-extraction">Default results extraction<a class="anchor" aria-label="anchor" href="#default-results-extraction"></a>
</h2>
<p>At this point, we could extract the taxa shared across all methods
using the function bake() to define a default consensus strategy and
then cool() to extract the results.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Default DA taxa results</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/bake.html">bake</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/cool.html">cool</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 26 × 2</span></span></span>
<span><span class="co">#&gt;    taxa_id taxa                        </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Otu_44  Gordonibacter_pamelaeae     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Otu_45  Slackia_isoflavoniconvertens</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Otu_48  Bacteroides_clarus          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Otu_63  Bacteroides_plebeius        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Otu_69  Bacteroides_sp_CAG_530      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Otu_78  Bacteroides_uniformis       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Otu_82  Barnesiella_intestinihominis</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Otu_88  Odoribacter_splanchnicus    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Otu_96  Prevotella_copri            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Otu_101 Prevotella_sp_885           </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 16 more rows</span></span></span></code></pre></div>
<p>However, ‘dar’ allows for complex consensus strategies based on the
obtained results. To that end, the user has access to different
functions to graphically represent different types of information. This
feature allows for a more in-depth analysis of the results and a better
understanding of the underlying patterns in the data.</p>
</div>
<div class="section level2">
<h2 id="exploration-for-consensus-strategie-definition">Exploration for consensus strategie definition<a class="anchor" aria-label="anchor" href="#exploration-for-consensus-strategie-definition"></a>
</h2>
<p>For example, intersection_plt() gives an overview of the overlaps
between methods by creating an upSet plot. In our case, this function
has shown that 25 taxa are shared across all the methods used. However,
by changing the consensus strategy and using a lower count_cutoff value,
we can see that the number of shared taxa increases from 25 to 80.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Intersection plot </span></span>
<span><span class="fu"><a href="../reference/intersection_plt.html">intersection_plt</a></span><span class="op">(</span><span class="va">da_results</span>, ordered_by <span class="op">=</span> <span class="st">"degree"</span>, font_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;"></p>
<p>In addition to the intersection_plt() function, ‘dar’ also has the
function exclusion_plt() which provides information about the number of
OTUs shared between methods. This function allows to identify the OTUs
that are specific to each method and also the ones that are not shared
among any method.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Exclusion plot </span></span>
<span><span class="fu"><a href="../reference/exclusion_plt.html">exclusion_plt</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span></span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;"></p>
<p>Besides to the previously mentioned functions, ‘dar’ also includes
the function corr_heatmap(), which allows for visualization of the
overlap of significant OTUs between tested methods. This function can
provide similar information to the previous plots, but in some cases it
may be easier to interpret. Using this function with our data set, we
can observe that although most of the tested methods return similar
results, DESeq2 and ANCOM have a lower correlation with the other
methods. This result can also be observed in the plot generated by the
intersection_plt() function, providing a more comprehensive view of the
results.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Correlation heatmap</span></span>
<span><span class="va">corr_heat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corr_heatmap.html">corr_heatmap</a></span><span class="op">(</span><span class="va">da_results</span>, font_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> </span>
<span><span class="co">#&gt; Warning: Using one column matrices in `filter()` was deprecated in dplyr 1.1.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use one dimensional logical vectors instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">dplyr</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/tidyverse/dplyr/issues&gt;</span>.</span></span>
<span><span class="va">corr_heat</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-f326911aef541fbb4c21" style="width:768px;height:576px;"></div>
<script type="application/json" data-for="htmlwidget-f326911aef541fbb4c21">{"x":{"data":[{"x":[4.859375,2.21875,null,2.21875,2.21875,null,2.21875,1,null,1,1,null,2.21875,3.4375,null,3.4375,3.4375,null,3.4375,2,null,2,2,null,3.4375,4.875,null,4.875,4.875,null,4.875,3.75,null,3.75,3.75,null,3.75,3,null,3,3,null,3.75,4.5,null,4.5,4.5,null,4.5,4,null,4,4,null,4.5,5,null,5,5,null,4.875,6,null,6,6,null,4.859375,7.5,null,7.5,7.5,null,7.5,7,null,7,7,null,7.5,8,null,8,8],"y":[0.921692882608251,0.921692882608251,null,0.921692882608251,0.437083984869691,null,0.437083984869691,0.437083984869691,null,0.437083984869691,0,null,0.437083984869691,0.437083984869691,null,0.437083984869691,0.350150487625927,null,0.350150487625927,0.350150487625927,null,0.350150487625927,0,null,0.350150487625927,0.350150487625927,null,0.350150487625927,0.272003668188488,null,0.272003668188488,0.272003668188488,null,0.272003668188488,0.215037780849077,null,0.215037780849077,0.215037780849077,null,0.215037780849077,0,null,0.215037780849077,0.215037780849077,null,0.215037780849077,0.156760709160757,null,0.156760709160757,0.156760709160757,null,0.156760709160757,0,null,0.156760709160757,0.156760709160757,null,0.156760709160757,0,null,0.272003668188488,0.272003668188488,null,0.272003668188488,0,null,0.921692882608251,0.921692882608251,null,0.921692882608251,0.843712611655291,null,0.843712611655291,0.843712611655291,null,0.843712611655291,0,null,0.843712611655291,0.843712611655291,null,0.843712611655291,0],"text":["y: 0.922","y: 0.922",null,"y: 0.922","y: 0.922",null,"y: 0.437","y: 0.437",null,"y: 0.437","y: 0.437",null,"y: 0.437","y: 0.437",null,"y: 0.437","y: 0.437",null,"y: 0.350","y: 0.350",null,"y: 0.350","y: 0.350",null,"y: 0.350","y: 0.350",null,"y: 0.350","y: 0.350",null,"y: 0.272","y: 0.272",null,"y: 0.272","y: 0.272",null,"y: 0.215","y: 0.215",null,"y: 0.215","y: 0.215",null,"y: 0.215","y: 0.215",null,"y: 0.215","y: 0.215",null,"y: 0.157","y: 0.157",null,"y: 0.157","y: 0.157",null,"y: 0.157","y: 0.157",null,"y: 0.157","y: 0.157",null,"y: 0.272","y: 0.272",null,"y: 0.272","y: 0.272",null,"y: 0.922","y: 0.922",null,"y: 0.922","y: 0.922",null,"y: 0.844","y: 0.844",null,"y: 0.844","y: 0.844",null,"y: 0.844","y: 0.844",null,"y: 0.844","y: 0.844"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.859375,2.21875,1,3.4375,2,4.875,3.75,3,4.5,4,5,6,7.5,7,8],"y":[0.921692882608251,0.437083984869691,0,0.350150487625927,0,0.272003668188488,0.215037780849077,0,0.156760709160757,0,0,0,0.843712611655291,0,0],"text":["y: 0.922","y: 0.437","y: 0.000","y: 0.350","y: 0.000","y: 0.272","y: 0.215","y: 0.000","y: 0.157","y: 0.000","y: 0.000","y: 0.000","y: 0.844","y: 0.000","y: 0.000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.88976377952756,"color":"transparent"}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"type":"scatter","mode":"markers","marker":{"color":"rgba(44,160,44,1)","line":{"color":"rgba(44,160,44,1)"}},"error_y":{"color":"rgba(44,160,44,1)"},"error_x":{"color":"rgba(44,160,44,1)"},"line":{"color":"rgba(44,160,44,1)"},"xaxis":"x2","yaxis":"y","frame":null},{"x":[1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,3,3,3,3,3,3,3,3,4,4,4,4,4,4,4,4,5,5,5,5,5,5,5,5,6,6,6,6,6,6,6,6,7,7,7,7,7,7,7,7,8,8,8,8,8,8,8,8],"y":[1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8],"text":["row: metagenomeseq__Pain_aux_raisins<br>column: metagenomeseq__Pain_aux_raisins<br>value: 1.0000 <br> Point size :  1.0000","row: maaslin__Nazook<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.7816 <br> Point size :  0.7816","row: corncob__Paste<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.7299 <br> Point size :  0.7299","row: lefse__Filo<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.7414 <br> Point size :  0.7414","row: wilcox__Cannoli_siciliani<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.7989 <br> Point size :  0.7989","row: aldex__Eccles_cake<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.7701 <br> Point size :  0.7701","row: ancom__Roti_john<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.5805 <br> Point size :  0.5805","row: deseq__Öçpoçmaq<br>column: metagenomeseq__Pain_aux_raisins<br>value: 0.4828 <br> Point size :  0.4828","row: metagenomeseq__Pain_aux_raisins<br>column: maaslin__Nazook<br>value: 0.7816 <br> Point size :  0.7816","row: maaslin__Nazook<br>column: maaslin__Nazook<br>value: 1.0000 <br> Point size :  1.0000","row: corncob__Paste<br>column: maaslin__Nazook<br>value: 0.8678 <br> Point size :  0.8678","row: lefse__Filo<br>column: maaslin__Nazook<br>value: 0.8448 <br> Point size :  0.8448","row: wilcox__Cannoli_siciliani<br>column: maaslin__Nazook<br>value: 0.8563 <br> Point size :  0.8563","row: aldex__Eccles_cake<br>column: maaslin__Nazook<br>value: 0.7931 <br> Point size :  0.7931","row: ancom__Roti_john<br>column: maaslin__Nazook<br>value: 0.5115 <br> Point size :  0.5115","row: deseq__Öçpoçmaq<br>column: maaslin__Nazook<br>value: 0.6207 <br> Point size :  0.6207","row: metagenomeseq__Pain_aux_raisins<br>column: corncob__Paste<br>value: 0.7299 <br> Point size :  0.7299","row: maaslin__Nazook<br>column: corncob__Paste<br>value: 0.8678 <br> Point size :  0.8678","row: corncob__Paste<br>column: corncob__Paste<br>value: 1.0000 <br> Point size :  1.0000","row: lefse__Filo<br>column: corncob__Paste<br>value: 0.8506 <br> Point size :  0.8506","row: wilcox__Cannoli_siciliani<br>column: corncob__Paste<br>value: 0.8966 <br> Point size :  0.8966","row: aldex__Eccles_cake<br>column: corncob__Paste<br>value: 0.8333 <br> Point size :  0.8333","row: ancom__Roti_john<br>column: corncob__Paste<br>value: 0.6322 <br> Point size :  0.6322","row: deseq__Öçpoçmaq<br>column: corncob__Paste<br>value: 0.5805 <br> Point size :  0.5805","row: metagenomeseq__Pain_aux_raisins<br>column: lefse__Filo<br>value: 0.7414 <br> Point size :  0.7414","row: maaslin__Nazook<br>column: lefse__Filo<br>value: 0.8448 <br> Point size :  0.8448","row: corncob__Paste<br>column: lefse__Filo<br>value: 0.8506 <br> Point size :  0.8506","row: lefse__Filo<br>column: lefse__Filo<br>value: 1.0000 <br> Point size :  1.0000","row: wilcox__Cannoli_siciliani<br>column: lefse__Filo<br>value: 0.9195 <br> Point size :  0.9195","row: aldex__Eccles_cake<br>column: lefse__Filo<br>value: 0.8218 <br> Point size :  0.8218","row: ancom__Roti_john<br>column: lefse__Filo<br>value: 0.6207 <br> Point size :  0.6207","row: deseq__Öçpoçmaq<br>column: lefse__Filo<br>value: 0.5920 <br> Point size :  0.5920","row: metagenomeseq__Pain_aux_raisins<br>column: wilcox__Cannoli_siciliani<br>value: 0.7989 <br> Point size :  0.7989","row: maaslin__Nazook<br>column: wilcox__Cannoli_siciliani<br>value: 0.8563 <br> Point size :  0.8563","row: corncob__Paste<br>column: wilcox__Cannoli_siciliani<br>value: 0.8966 <br> Point size :  0.8966","row: lefse__Filo<br>column: wilcox__Cannoli_siciliani<br>value: 0.9195 <br> Point size :  0.9195","row: wilcox__Cannoli_siciliani<br>column: wilcox__Cannoli_siciliani<br>value: 1.0000 <br> Point size :  1.0000","row: aldex__Eccles_cake<br>column: wilcox__Cannoli_siciliani<br>value: 0.8908 <br> Point size :  0.8908","row: ancom__Roti_john<br>column: wilcox__Cannoli_siciliani<br>value: 0.6552 <br> Point size :  0.6552","row: deseq__Öçpoçmaq<br>column: wilcox__Cannoli_siciliani<br>value: 0.5805 <br> Point size :  0.5805","row: metagenomeseq__Pain_aux_raisins<br>column: aldex__Eccles_cake<br>value: 0.7701 <br> Point size :  0.7701","row: maaslin__Nazook<br>column: aldex__Eccles_cake<br>value: 0.7931 <br> Point size :  0.7931","row: corncob__Paste<br>column: aldex__Eccles_cake<br>value: 0.8333 <br> Point size :  0.8333","row: lefse__Filo<br>column: aldex__Eccles_cake<br>value: 0.8218 <br> Point size :  0.8218","row: wilcox__Cannoli_siciliani<br>column: aldex__Eccles_cake<br>value: 0.8908 <br> Point size :  0.8908","row: aldex__Eccles_cake<br>column: aldex__Eccles_cake<br>value: 1.0000 <br> Point size :  1.0000","row: ancom__Roti_john<br>column: aldex__Eccles_cake<br>value: 0.6954 <br> Point size :  0.6954","row: deseq__Öçpoçmaq<br>column: aldex__Eccles_cake<br>value: 0.6092 <br> Point size :  0.6092","row: metagenomeseq__Pain_aux_raisins<br>column: ancom__Roti_john<br>value: 0.5805 <br> Point size :  0.5805","row: maaslin__Nazook<br>column: ancom__Roti_john<br>value: 0.5115 <br> Point size :  0.5115","row: corncob__Paste<br>column: ancom__Roti_john<br>value: 0.6322 <br> Point size :  0.6322","row: lefse__Filo<br>column: ancom__Roti_john<br>value: 0.6207 <br> Point size :  0.6207","row: wilcox__Cannoli_siciliani<br>column: ancom__Roti_john<br>value: 0.6552 <br> Point size :  0.6552","row: aldex__Eccles_cake<br>column: ancom__Roti_john<br>value: 0.6954 <br> Point size :  0.6954","row: ancom__Roti_john<br>column: ancom__Roti_john<br>value: 1.0000 <br> Point size :  1.0000","row: deseq__Öçpoçmaq<br>column: ancom__Roti_john<br>value: 0.4195 <br> Point size :  0.4195","row: metagenomeseq__Pain_aux_raisins<br>column: deseq__Öçpoçmaq<br>value: 0.4828 <br> Point size :  0.4828","row: maaslin__Nazook<br>column: deseq__Öçpoçmaq<br>value: 0.6207 <br> Point size :  0.6207","row: corncob__Paste<br>column: deseq__Öçpoçmaq<br>value: 0.5805 <br> Point size :  0.5805","row: lefse__Filo<br>column: deseq__Öçpoçmaq<br>value: 0.5920 <br> Point size :  0.5920","row: wilcox__Cannoli_siciliani<br>column: deseq__Öçpoçmaq<br>value: 0.5805 <br> Point size :  0.5805","row: aldex__Eccles_cake<br>column: deseq__Öçpoçmaq<br>value: 0.6092 <br> Point size :  0.6092","row: ancom__Roti_john<br>column: deseq__Öçpoçmaq<br>value: 0.4195 <br> Point size :  0.4195","row: deseq__Öçpoçmaq<br>column: deseq__Öçpoçmaq<br>value: 1.0000 <br> Point size :  1.0000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":["rgba(165,0,38,1)","rgba(246,122,72,1)","rgba(251,155,88,1)","rgba(250,148,84,1)","rgba(244,110,67,1)","rgba(247,130,76,1)","rgba(255,230,149,1)","rgba(249,252,182,1)","rgba(246,122,72,1)","rgba(165,0,38,1)","rgba(225,70,48,1)","rgba(231,84,54,1)","rgba(228,77,51,1)","rgba(245,114,69,1)","rgba(255,251,185,1)","rgba(254,214,130,1)","rgba(251,155,88,1)","rgba(225,70,48,1)","rgba(165,0,38,1)","rgba(230,81,53,1)","rgba(216,51,40,1)","rgba(235,91,57,1)","rgba(254,208,125,1)","rgba(255,230,149,1)","rgba(250,148,84,1)","rgba(231,84,54,1)","rgba(230,81,53,1)","rgba(165,0,38,1)","rgba(205,41,39,1)","rgba(238,97,61,1)","rgba(254,214,130,1)","rgba(254,226,143,1)","rgba(244,110,67,1)","rgba(228,77,51,1)","rgba(216,51,40,1)","rgba(205,41,39,1)","rgba(165,0,38,1)","rgba(218,55,41,1)","rgba(254,197,116,1)","rgba(255,230,149,1)","rgba(247,130,76,1)","rgba(245,114,69,1)","rgba(235,91,57,1)","rgba(238,97,61,1)","rgba(218,55,41,1)","rgba(165,0,38,1)","rgba(253,176,99,1)","rgba(254,219,135,1)","rgba(255,230,149,1)","rgba(255,251,185,1)","rgba(254,208,125,1)","rgba(254,214,130,1)","rgba(254,197,116,1)","rgba(253,176,99,1)","rgba(165,0,38,1)","rgba(225,242,149,1)","rgba(249,252,182,1)","rgba(254,214,130,1)","rgba(255,230,149,1)","rgba(254,226,143,1)","rgba(255,230,149,1)","rgba(254,219,135,1)","rgba(225,242,149,1)","rgba(165,0,38,1)"],"opacity":1,"size":[22.6771653543307,18.7046227118141,17.5974806424973,17.851042215549,19.0558494752258,18.4658057808239,13.7295909942278,10.0160598659075,18.7046227118141,22.6771653543307,20.3866403471573,19.955213032423,20.1723460429228,18.9396780408486,11.301068524422,14.9040366697957,17.5974806424973,20.3866403471573,22.6771653543307,20.0641414380877,20.910652403502,19.735125423805,15.2174644476143,13.7295909942278,17.851042215549,19.955213032423,20.0641414380877,22.6771653543307,21.3185934708832,19.5119592182075,14.9040366697957,14.0788216938816,19.0558494752258,20.1723460429228,20.910652403502,21.3185934708832,22.6771653543307,20.8071401274738,15.8198678736012,13.7295909942278,18.4658057808239,18.9396780408486,19.735125423805,19.5119592182075,20.8071401274738,22.6771653543307,16.8072186622812,14.5815183775681,13.7295909942278,11.301068524422,15.2174644476143,14.9040366697957,15.8198678736012,16.8072186622812,22.6771653543307,3.77952755905512,10.0160598659075,14.9040366697957,13.7295909942278,14.0788216938816,13.7295909942278,14.5815183775681,3.77952755905512,22.6771653543307],"symbol":"circle","line":{"width":1.88976377952756,"color":["rgba(165,0,38,1)","rgba(246,122,72,1)","rgba(251,155,88,1)","rgba(250,148,84,1)","rgba(244,110,67,1)","rgba(247,130,76,1)","rgba(255,230,149,1)","rgba(249,252,182,1)","rgba(246,122,72,1)","rgba(165,0,38,1)","rgba(225,70,48,1)","rgba(231,84,54,1)","rgba(228,77,51,1)","rgba(245,114,69,1)","rgba(255,251,185,1)","rgba(254,214,130,1)","rgba(251,155,88,1)","rgba(225,70,48,1)","rgba(165,0,38,1)","rgba(230,81,53,1)","rgba(216,51,40,1)","rgba(235,91,57,1)","rgba(254,208,125,1)","rgba(255,230,149,1)","rgba(250,148,84,1)","rgba(231,84,54,1)","rgba(230,81,53,1)","rgba(165,0,38,1)","rgba(205,41,39,1)","rgba(238,97,61,1)","rgba(254,214,130,1)","rgba(254,226,143,1)","rgba(244,110,67,1)","rgba(228,77,51,1)","rgba(216,51,40,1)","rgba(205,41,39,1)","rgba(165,0,38,1)","rgba(218,55,41,1)","rgba(254,197,116,1)","rgba(255,230,149,1)","rgba(247,130,76,1)","rgba(245,114,69,1)","rgba(235,91,57,1)","rgba(238,97,61,1)","rgba(218,55,41,1)","rgba(165,0,38,1)","rgba(253,176,99,1)","rgba(254,219,135,1)","rgba(255,230,149,1)","rgba(255,251,185,1)","rgba(254,208,125,1)","rgba(254,214,130,1)","rgba(254,197,116,1)","rgba(253,176,99,1)","rgba(165,0,38,1)","rgba(225,242,149,1)","rgba(249,252,182,1)","rgba(254,214,130,1)","rgba(255,230,149,1)","rgba(254,226,143,1)","rgba(255,230,149,1)","rgba(254,219,135,1)","rgba(225,242,149,1)","rgba(165,0,38,1)"]}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1],"y":[1],"name":"99_0e1c4cd0ab8b9a97b06de41b32373fad","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#006837"],[0.00334448160535117,"#016A38"],[0.00668896321070234,"#016B39"],[0.0100334448160535,"#026D39"],[0.0133779264214047,"#036E3A"],[0.0167224080267559,"#04703B"],[0.020066889632107,"#04713C"],[0.0234113712374582,"#05733D"],[0.0267558528428094,"#06753E"],[0.0301003344481605,"#07763E"],[0.0334448160535117,"#08783F"],[0.0367892976588629,"#097940"],[0.040133779264214,"#0A7B41"],[0.0434782608695652,"#0B7C42"],[0.0468227424749164,"#0C7E43"],[0.0501672240802676,"#0D8043"],[0.0535117056856187,"#0E8144"],[0.0568561872909699,"#0F8345"],[0.0602006688963211,"#108546"],[0.0635451505016722,"#118647"],[0.0668896321070234,"#128848"],[0.0702341137123746,"#128948"],[0.0735785953177257,"#138B49"],[0.0769230769230769,"#148D4A"],[0.0802675585284281,"#158E4B"],[0.0836120401337793,"#16904C"],[0.0869565217391304,"#17924D"],[0.0903010033444816,"#18934E"],[0.0936454849498328,"#18954E"],[0.096989966555184,"#19974F"],[0.100334448160535,"#1A9850"],[0.103678929765886,"#1F9951"],[0.107023411371237,"#229B51"],[0.110367892976589,"#269C52"],[0.11371237458194,"#299D53"],[0.117056856187291,"#2C9E53"],[0.120401337792642,"#2F9F54"],[0.123745819397993,"#32A154"],[0.127090301003344,"#35A255"],[0.130434782608696,"#38A356"],[0.133779264214047,"#3AA456"],[0.137123745819398,"#3DA657"],[0.140468227424749,"#3FA758"],[0.1438127090301,"#42A858"],[0.147157190635451,"#44A959"],[0.150501672240803,"#46AB5A"],[0.153846153846154,"#49AC5A"],[0.157190635451505,"#4BAD5B"],[0.160535117056856,"#4DAE5B"],[0.163879598662207,"#4FB05C"],[0.167224080267559,"#52B15D"],[0.17056856187291,"#54B25D"],[0.173913043478261,"#56B35E"],[0.177257525083612,"#58B55F"],[0.180602006688963,"#5AB65F"],[0.183946488294314,"#5CB760"],[0.187290969899666,"#5EB861"],[0.190635451505017,"#60B961"],[0.193979933110368,"#62BB62"],[0.197324414715719,"#64BC62"],[0.20066889632107,"#66BD63"],[0.204013377926421,"#69BE63"],[0.207357859531773,"#6BBF64"],[0.210702341137124,"#6DC064"],[0.214046822742475,"#70C164"],[0.217391304347826,"#72C264"],[0.220735785953177,"#74C364"],[0.224080267558528,"#76C465"],[0.22742474916388,"#79C565"],[0.230769230769231,"#7BC665"],[0.234113712374582,"#7DC765"],[0.237458193979933,"#7FC766"],[0.240802675585284,"#81C866"],[0.244147157190635,"#83C966"],[0.247491638795987,"#86CA66"],[0.250836120401338,"#88CB67"],[0.254180602006689,"#8ACC67"],[0.25752508361204,"#8CCD67"],[0.260869565217391,"#8ECE67"],[0.264214046822742,"#90CF68"],[0.267558528428094,"#92D068"],[0.270903010033445,"#94D168"],[0.274247491638796,"#96D268"],[0.277591973244147,"#98D368"],[0.280936454849498,"#9AD469"],[0.284280936454849,"#9CD569"],[0.287625418060201,"#9FD669"],[0.290969899665552,"#A1D669"],[0.294314381270903,"#A3D76A"],[0.297658862876254,"#A5D86A"],[0.301003344481605,"#A7D96A"],[0.304347826086957,"#A8DA6B"],[0.307692307692308,"#AADB6D"],[0.311036789297659,"#ACDB6E"],[0.31438127090301,"#ADDC6F"],[0.317725752508361,"#AFDD70"],[0.321070234113712,"#B1DE71"],[0.324414715719064,"#B3DE72"],[0.327759197324415,"#B4DF73"],[0.331103678929766,"#B6E074"],[0.334448160535117,"#B8E175"],[0.337792642140468,"#B9E176"],[0.341137123745819,"#BBE278"],[0.344481605351171,"#BDE379"],[0.347826086956522,"#BFE47A"],[0.351170568561873,"#C0E47B"],[0.354515050167224,"#C2E57C"],[0.357859531772575,"#C4E67D"],[0.361204013377926,"#C5E77E"],[0.364548494983278,"#C7E77F"],[0.367892976588629,"#C9E880"],[0.37123745819398,"#CAE981"],[0.374581939799331,"#CCE983"],[0.377926421404682,"#CEEA84"],[0.381270903010033,"#D0EB85"],[0.384615384615385,"#D1EC86"],[0.387959866220736,"#D3EC87"],[0.391304347826087,"#D5ED88"],[0.394648829431438,"#D6EE89"],[0.397993311036789,"#D8EF8A"],[0.40133779264214,"#DAEF8C"],[0.404682274247492,"#DBF08D"],[0.408026755852843,"#DCF08F"],[0.411371237458194,"#DDF191"],[0.414715719063545,"#DFF193"],[0.418060200668896,"#E0F294"],[0.421404682274247,"#E1F296"],[0.424749163879599,"#E3F398"],[0.42809364548495,"#E4F49A"],[0.431438127090301,"#E5F49B"],[0.434782608695652,"#E6F59D"],[0.438127090301003,"#E8F59F"],[0.441471571906354,"#E9F6A1"],[0.444816053511706,"#EAF6A2"],[0.448160535117057,"#EBF7A4"],[0.451505016722408,"#EDF7A6"],[0.454849498327759,"#EEF8A8"],[0.45819397993311,"#EFF8A9"],[0.461538461538462,"#F1F9AB"],[0.464882943143813,"#F2F9AD"],[0.468227424749164,"#F3FAAF"],[0.471571906354515,"#F4FAB0"],[0.474916387959866,"#F6FBB2"],[0.478260869565217,"#F7FCB4"],[0.481605351170569,"#F8FCB5"],[0.48494983277592,"#F9FDB7"],[0.488294314381271,"#FBFDB9"],[0.491638795986622,"#FCFEBB"],[0.494983277591973,"#FDFEBC"],[0.498327759197324,"#FEFFBE"],[0.501672240802676,"#FFFEBE"],[0.505016722408027,"#FFFDBC"],[0.508361204013378,"#FFFCBB"],[0.511705685618729,"#FFFBB9"],[0.51505016722408,"#FFFAB7"],[0.518394648829431,"#FFF9B5"],[0.521739130434783,"#FFF8B4"],[0.525083612040134,"#FFF7B2"],[0.528428093645485,"#FFF6B0"],[0.531772575250836,"#FFF5AE"],[0.535117056856187,"#FFF4AD"],[0.538461538461538,"#FFF3AB"],[0.54180602006689,"#FFF2A9"],[0.545150501672241,"#FFF1A7"],[0.548494983277592,"#FFF0A6"],[0.551839464882943,"#FFEFA4"],[0.555183946488294,"#FFEEA2"],[0.558528428093645,"#FFEDA1"],[0.561872909698997,"#FFEC9F"],[0.565217391304348,"#FFEB9D"],[0.568561872909699,"#FFEA9B"],[0.57190635451505,"#FFE99A"],[0.575250836120401,"#FFE898"],[0.578595317725752,"#FFE796"],[0.581939799331104,"#FFE694"],[0.585284280936455,"#FEE593"],[0.588628762541806,"#FEE391"],[0.591973244147157,"#FEE28F"],[0.595317725752508,"#FEE18D"],[0.598662207357859,"#FEE08C"],[0.602006688963211,"#FEDF8A"],[0.605351170568562,"#FEDD89"],[0.608695652173913,"#FEDC87"],[0.612040133779264,"#FEDA86"],[0.615384615384615,"#FED884"],[0.618729096989967,"#FED783"],[0.622073578595318,"#FED582"],[0.625418060200669,"#FED380"],[0.62876254180602,"#FED27F"],[0.632107023411371,"#FED07D"],[0.635451505016722,"#FECE7C"],[0.638795986622074,"#FECD7B"],[0.642140468227425,"#FECB79"],[0.645484949832776,"#FEC978"],[0.648829431438127,"#FEC876"],[0.652173913043478,"#FEC675"],[0.655518394648829,"#FEC473"],[0.658862876254181,"#FEC372"],[0.662207357859532,"#FEC171"],[0.665551839464883,"#FEBF6F"],[0.668896321070234,"#FEBE6E"],[0.672240802675585,"#FEBC6C"],[0.675585284280936,"#FEBA6B"],[0.678929765886288,"#FEB96A"],[0.682274247491639,"#FEB768"],[0.68561872909699,"#FEB567"],[0.688963210702341,"#FDB466"],[0.692307692307692,"#FDB264"],[0.695652173913043,"#FDB063"],[0.698996655518395,"#FDAF61"],[0.702341137123746,"#FDAD60"],[0.705685618729097,"#FDAB5F"],[0.709030100334448,"#FCA85E"],[0.712374581939799,"#FCA65D"],[0.71571906354515,"#FCA45C"],[0.719063545150502,"#FCA25B"],[0.722408026755853,"#FCA05A"],[0.725752508361204,"#FB9E59"],[0.729096989966555,"#FB9C58"],[0.732441471571906,"#FB9A57"],[0.735785953177257,"#FB9856"],[0.739130434782609,"#FA9655"],[0.74247491638796,"#FA9354"],[0.745819397993311,"#FA9153"],[0.749163879598662,"#F98F52"],[0.752508361204013,"#F98D51"],[0.755852842809364,"#F98B50"],[0.759197324414716,"#F8894F"],[0.762541806020067,"#F8874E"],[0.765886287625418,"#F8844D"],[0.769230769230769,"#F7824C"],[0.77257525083612,"#F7804B"],[0.775919732441472,"#F77E4A"],[0.779264214046823,"#F67B49"],[0.782608695652174,"#F67948"],[0.785953177257525,"#F67747"],[0.789297658862876,"#F57546"],[0.792642140468227,"#F57245"],[0.795986622073579,"#F47044"],[0.79933110367893,"#F46D43"],[0.802675585284281,"#F36C42"],[0.806020066889632,"#F26A41"],[0.809364548494983,"#F16840"],[0.812709030100334,"#F0663F"],[0.816053511705686,"#EF643E"],[0.819397993311037,"#EF633D"],[0.822742474916388,"#EE613C"],[0.826086956521739,"#ED5F3B"],[0.82943143812709,"#EC5D3A"],[0.832775919732441,"#EB5B3A"],[0.836120401337793,"#EA5939"],[0.839464882943144,"#E95738"],[0.842809364548495,"#E85537"],[0.846153846153846,"#E75336"],[0.849498327759197,"#E65135"],[0.852842809364548,"#E54F34"],[0.8561872909699,"#E44D33"],[0.859531772575251,"#E34B32"],[0.862876254180602,"#E24931"],[0.866220735785953,"#E14730"],[0.869565217391304,"#E0452F"],[0.872909698996655,"#DF432E"],[0.876254180602007,"#DE412D"],[0.879598662207358,"#DD3F2C"],[0.882943143812709,"#DC3C2C"],[0.88628762541806,"#DB3A2B"],[0.889632107023411,"#DA382A"],[0.892976588628763,"#D93529"],[0.896321070234114,"#D83328"],[0.899665551839465,"#D73027"],[0.903010033444816,"#D52F27"],[0.906354515050167,"#D42E27"],[0.909698996655518,"#D22C27"],[0.91304347826087,"#D02B27"],[0.916387959866221,"#CF2A27"],[0.919732441471572,"#CD2927"],[0.923076923076923,"#CB2727"],[0.926421404682274,"#CA2627"],[0.929765886287625,"#C82527"],[0.933110367892977,"#C62427"],[0.936454849498328,"#C52227"],[0.939799331103679,"#C32127"],[0.94314381270903,"#C11F27"],[0.946488294314381,"#BF1E27"],[0.949832775919732,"#BE1D27"],[0.953177257525084,"#BC1B27"],[0.956521739130435,"#BA1A27"],[0.959866220735786,"#B91827"],[0.963210702341137,"#B71627"],[0.966555183946488,"#B51527"],[0.969899665551839,"#B41327"],[0.973244147157191,"#B21127"],[0.976588628762542,"#B10F27"],[0.979933110367893,"#AF0D26"],[0.983277591973244,"#AD0B26"],[0.986622073578595,"#AC0926"],[0.989966555183946,"#AA0726"],[0.993311036789298,"#A80426"],[0.996655518394649,"#A70226"],[1,"#A50026"]],"colorbar":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"thickness":23.04,"title":"value","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["0.00","0.25","0.50","0.75","1.00"],"tickvals":[0,0.25,0.5,0.75,1],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y2","frame":null},{"x":[0.921692882608251,0.921692882608251,null,0.921692882608251,0.437083984869691,null,0.437083984869691,0.437083984869691,null,0.437083984869691,0,null,0.437083984869691,0.437083984869691,null,0.437083984869691,0.350150487625927,null,0.350150487625927,0.350150487625927,null,0.350150487625927,0,null,0.350150487625927,0.350150487625927,null,0.350150487625927,0.272003668188488,null,0.272003668188488,0.272003668188488,null,0.272003668188488,0.215037780849077,null,0.215037780849077,0.215037780849077,null,0.215037780849077,0,null,0.215037780849077,0.215037780849077,null,0.215037780849077,0.156760709160757,null,0.156760709160757,0.156760709160757,null,0.156760709160757,0,null,0.156760709160757,0.156760709160757,null,0.156760709160757,0,null,0.272003668188488,0.272003668188488,null,0.272003668188488,0,null,0.921692882608251,0.921692882608251,null,0.921692882608251,0.843712611655291,null,0.843712611655291,0.843712611655291,null,0.843712611655291,0,null,0.843712611655291,0.843712611655291,null,0.843712611655291,0],"y":[4.859375,2.21875,null,2.21875,2.21875,null,2.21875,1,null,1,1,null,2.21875,3.4375,null,3.4375,3.4375,null,3.4375,2,null,2,2,null,3.4375,4.875,null,4.875,4.875,null,4.875,3.75,null,3.75,3.75,null,3.75,3,null,3,3,null,3.75,4.5,null,4.5,4.5,null,4.5,4,null,4,4,null,4.5,5,null,5,5,null,4.875,6,null,6,6,null,4.859375,7.5,null,7.5,7.5,null,7.5,7,null,7,7,null,7.5,8,null,8,8],"text":["y: 0.922","y: 0.922",null,"y: 0.922","y: 0.922",null,"y: 0.437","y: 0.437",null,"y: 0.437","y: 0.437",null,"y: 0.437","y: 0.437",null,"y: 0.437","y: 0.437",null,"y: 0.350","y: 0.350",null,"y: 0.350","y: 0.350",null,"y: 0.350","y: 0.350",null,"y: 0.350","y: 0.350",null,"y: 0.272","y: 0.272",null,"y: 0.272","y: 0.272",null,"y: 0.215","y: 0.215",null,"y: 0.215","y: 0.215",null,"y: 0.215","y: 0.215",null,"y: 0.215","y: 0.215",null,"y: 0.157","y: 0.157",null,"y: 0.157","y: 0.157",null,"y: 0.157","y: 0.157",null,"y: 0.157","y: 0.157",null,"y: 0.272","y: 0.272",null,"y: 0.272","y: 0.272",null,"y: 0.922","y: 0.922",null,"y: 0.922","y: 0.922",null,"y: 0.844","y: 0.844",null,"y: 0.844","y: 0.844",null,"y: 0.844","y: 0.844",null,"y: 0.844","y: 0.844"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.921692882608251,0.437083984869691,0,0.350150487625927,0,0.272003668188488,0.215037780849077,0,0.156760709160757,0,0,0,0.843712611655291,0,0],"y":[4.859375,2.21875,1,3.4375,2,4.875,3.75,3,4.5,4,5,6,7.5,7,8],"text":["y: 0.922","y: 0.437","y: 0.000","y: 0.350","y: 0.000","y: 0.272","y: 0.215","y: 0.000","y: 0.157","y: 0.000","y: 0.000","y: 0.000","y: 0.844","y: 0.000","y: 0.000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.88976377952756,"color":"transparent"}},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"xaxis":{"domain":[0,0.8],"automargin":true,"type":"linear","autorange":false,"range":[0.4,8.6],"tickmode":"array","ticktext":["metagenomeseq__Pain_aux_raisins","maaslin__Nazook","corncob__Paste","lefse__Filo","wilcox__Cannoli_siciliani","aldex__Eccles_cake","ancom__Roti_john","deseq__Öçpoçmaq"],"tickvals":[1,2,3,4,5,6,7,8],"categoryorder":"array","categoryarray":["metagenomeseq__Pain_aux_raisins","maaslin__Nazook","corncob__Paste","lefse__Filo","wilcox__Cannoli_siciliani","aldex__Eccles_cake","ancom__Roti_john","deseq__Öçpoçmaq"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"Arial","size":13.2835201328352},"tickangle":-45,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"xaxis2":{"domain":[0.8,1],"automargin":true,"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"y2","type":"linear","autorange":false,"range":[0,0.921692882608251],"tickmode":"array","ticktext":["0.00","0.25","0.50","0.75"],"tickvals":[0,0.25,0.5,0.75],"categoryorder":"array","categoryarray":["0.00","0.25","0.50","0.75"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":{"text":"","font":{"color":null,"family":null,"size":0}},"hoverformat":".2f"},"yaxis2":{"domain":[0,0.8],"automargin":true,"type":"linear","autorange":false,"range":[0.4,8.6],"tickmode":"array","ticktext":["metagenomeseq__Pain_aux_raisins","maaslin__Nazook","corncob__Paste","lefse__Filo","wilcox__Cannoli_siciliani","aldex__Eccles_cake","ancom__Roti_john","deseq__Öçpoçmaq"],"tickvals":[1,2,3,4,5,6,7,8],"categoryorder":"array","categoryarray":["metagenomeseq__Pain_aux_raisins","maaslin__Nazook","corncob__Paste","lefse__Filo","wilcox__Cannoli_siciliani","aldex__Eccles_cake","ancom__Roti_john","deseq__Öçpoçmaq"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"Arial","size":13.2835201328352},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"yaxis":{"domain":[0.8,1],"automargin":true,"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"x","type":"linear","autorange":false,"range":[0,0.921692882608251],"tickmode":"array","ticktext":["0.00","0.25","0.50","0.75"],"tickvals":[0,0.25,0.5,0.75],"categoryorder":"array","categoryarray":["0.00","0.25","0.50","0.75"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":{"text":"","font":{"color":null,"family":null,"size":0}},"hoverformat":".2f"},"annotations":[],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.8,"y0":0.8,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.8,"y0":0,"y1":0.8},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.8,"x1":1,"y0":0,"y1":0.8}],"images":[],"margin":{"t":0,"r":null,"b":247.073474470735,"l":247.073474470735},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"title":{"text":"Point size","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"y":1,"yanchor":"top"},"hovermode":"closest","barmode":"relative","paper_bgcolor":"rgba(255,255,255,1)","title":""},"attrs":{"35cc1260eab":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"scatter"},"35cc67b56138":{"colour":{},"shape":{},"size":{},"x":{},"y":{}},"35cc1081403":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20]},"35cc2018e1da":{"x":{},"y":{},"colour":{},"text":{},"size":{},"type":"scatter"},"35cc28521aa3":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"scatter"},"35cc3cd2d507":{"colour":{},"shape":{},"size":{},"x":{},"y":{}}},"source":"A","config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false,"displaylogo":false,"modeBarButtonsToRemove":["sendDataToCloud","select2d","lasso2d","autoScale2d","hoverClosestCartesian","hoverCompareCartesian","sendDataToCloud"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"subplot":true,"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p>Finally, ‘dar’ also includes the function mutual_plt(), which plots
the number of differential abundant features mutually found by a defined
number of methods, colored by the differential abundance direction and
separated by comparison. The resulting graph allows us to see that the
features detected correspond mainly to the comparisons between hts vs
msm and msm vs pwid. Additionally, the graph also allows us to observe
the direction of the effect; whether a specific OTU is enriched or
depleted for each comparison.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Mutual plot</span></span>
<span><span class="fu"><a href="../reference/mutual_plt.html">mutual_plt</a></span><span class="op">(</span><span class="va">da_results</span>, count_cutoff <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/steps_ids.html">steps_ids</a></span><span class="op">(</span><span class="va">da_results</span>, type <span class="op">=</span> <span class="st">"da"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="define-a-consesus-strategy-using-bake">Define a consesus strategy using bake<a class="anchor" aria-label="anchor" href="#define-a-consesus-strategy-using-bake"></a>
</h2>
<p>After visually inspecting the results from running all the
differential analysis methods on our data, we have the necessary
information to define a consensus strategy that fits our dataset. In our
case, since DESeq2 returns results that are quite different from the
other methods, we will exclude the results obtained by this method and
keep only the differentially abundant features that are present in all
of the other 7 methods.</p>
<p>Additionally, the bake() function allows to further refine the
consensus strategy through its parameters, such as count_cutoff, which
indicates the minimum number of methods in which an OTU must be present,
and weights, a named vector with the ponderation value for each method.
However, for simplicity, these parameters are not used in this
example.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract id for deseq step</span></span>
<span><span class="va">to_exclude</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/steps_ids.html">steps_ids</a></span><span class="op">(</span><span class="va">da_results</span>, type <span class="op">=</span> <span class="st">"da"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="va">.</span><span class="op">[</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"deseq"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Define consensus strategy</span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bake.html">bake</a></span><span class="op">(</span><span class="va">da_results</span>, exclude <span class="op">=</span> <span class="va">to_exclude</span>, <span class="op">)</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Results</span> ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">174</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> ancom__Roti_john <span style="color: #555555;">diff_taxa = 63</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> aldex__Eccles_cake <span style="color: #555555;">diff_taxa = 112</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> deseq__Öçpoçmaq <span style="color: #555555;">diff_taxa = 92</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> corncob__Paste <span style="color: #555555;">diff_taxa = 125</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> metagenomeseq__Pain_aux_raisins <span style="color: #555555;">diff_taxa = 136</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> maaslin__Nazook <span style="color: #555555;">diff_taxa = 148</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> lefse__Filo <span style="color: #555555;">diff_taxa = 121</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> wilcox__Cannoli_siciliani <span style="color: #555555;">diff_taxa = 123</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> 26 taxa are present in all tested methods </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Bakes:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> <span style="color: #0000BB; font-weight: bold;">1 -&gt;</span> <span style="color: #555555;">count_cutoff: NULL, weights: NULL, exclude: deseq__Öçpoçmaq, id: bake__Cherry_pie</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="extract-results">Extract results<a class="anchor" aria-label="anchor" href="#extract-results"></a>
</h2>
<p>To conclude, we can extract the final results using the cool()
function. This function takes a prep-recipe object and the ID of the
bake to be used as input (by default it is 1, but if you have multiple
consensus strategies, you can change it to extract the desired
results).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract results for bake id 1</span></span>
<span><span class="va">f_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cool.html">cool</a></span><span class="op">(</span><span class="va">da_results</span>, bake <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">f_results</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 81 × 2</span></span></span>
<span><span class="co">#&gt;    taxa_id taxa                             </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Otu_13  Bifidobacterium_bifidum          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Otu_19  Bifidobacterium_pseudocatenulatum</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Otu_40  Adlercreutzia_equolifaciens      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Otu_41  Asaccharobacter_celatus          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Otu_42  Eggerthella_lenta                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Otu_44  Gordonibacter_pamelaeae          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Otu_45  Slackia_isoflavoniconvertens     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Otu_46  Bacteroides_caccae               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Otu_47  Bacteroides_cellulosilyticus     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Otu_48  Bacteroides_clarus               </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 71 more rows</span></span></span></code></pre></div>
<p>To further visualize the results, the abundance_plt() function can be
utilized to visualize the differences in abundance of the differential
abundant taxa.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Ids for Bacteroide and Provotella species </span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">f_results</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">taxa</span>, <span class="st">"Bacteroi.*|Prevote.*"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">taxa_id</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## Abundance plot as boxplot</span></span>
<span><span class="fu"><a href="../reference/abundance_plt.html">abundance_plt</a></span><span class="op">(</span><span class="va">da_results</span>, taxa_ids <span class="op">=</span> <span class="va">ids</span>, type <span class="op">=</span> <span class="st">"boxplot"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="article_files/figure-html/unnamed-chunk-13-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  C.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    C.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-01-31</span></span>
<span><span class="co">#&gt;  pandoc   2.19.2 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package                 </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version     </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                      1.4-5        <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ade4                       1.7-20       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ALDEx2                     1.30.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ANCOMBC                    2.0.2        <span style="color: #555555;">2022-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  annotate                   1.76.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi              1.60.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ape                        5.6-2        <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ashr                       2.2-54       <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  assertthat                 0.2.1        <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  backports                  1.4.1        <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  base64enc                  0.1-3        <span style="color: #555555;">2015-07-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  beachmat                   2.14.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  beeswarm                   0.4.0        <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  biglm                      0.9-2.1      <span style="color: #555555;">2020-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Biobase                    2.58.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics               0.44.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors              1.16.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocParallel               1.32.5       <span style="color: #555555;">2022-12-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocSingular               1.14.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  biomformat                 1.26.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings                 2.66.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                        4.0.5        <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  bit64                      4.0.5        <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  bitops                     1.0-7        <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  blob                       1.2.3        <span style="color: #555555;">2022-04-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  boot                       1.3-28       <span style="color: #555555;">2021-05-03</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  brio                       1.1.3        <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  broom                      1.0.3        <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  bslib                      0.4.2        <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ca                         0.71.1       <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  cachem                     1.0.6        <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  callr                      3.7.3        <span style="color: #555555;">2022-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  car                        3.1-1        <span style="color: #555555;">2022-10-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  carData                    3.0-5        <span style="color: #555555;">2022-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  caTools                    1.18.2       <span style="color: #555555;">2021-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  cellranger                 1.1.0        <span style="color: #555555;">2016-07-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  checkmate                  2.1.0        <span style="color: #555555;">2022-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  class                      7.3-20       <span style="color: #555555;">2022-01-16</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  cli                        3.6.0        <span style="color: #555555;">2023-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  cluster                    2.1.4        <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  coda                       0.19-4       <span style="color: #555555;">2020-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  codetools                  0.2-18       <span style="color: #555555;">2020-11-04</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  coin                       1.4-2        <span style="color: #555555;">2021-10-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  colorspace                 2.1-0        <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  corncob                    0.3.1        <span style="color: #555555;">2022-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  crayon                     1.5.2        <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  crosstalk                  1.2.0        <span style="color: #555555;">2021-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  CVXR                       1.0-11       <span style="color: #555555;">2022-10-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  dar                      * 0.99.0       <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">#&gt;  data.table                 1.14.6       <span style="color: #555555;">2022-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DBI                        1.1.3        <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DECIPHER                   2.26.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  decontam                   1.18.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedArray               0.24.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedMatrixStats         1.20.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  deldir                     1.0-6        <span style="color: #555555;">2021-10-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  dendextend                 1.16.0       <span style="color: #555555;">2022-07-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DEoptimR                   1.0-11       <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  desc                       1.4.2        <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DescTools                  0.99.47      <span style="color: #555555;">2022-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DESeq2                     1.38.3       <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  detectseparation           0.3          <span style="color: #555555;">2022-08-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  devtools                   2.4.5        <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  digest                     0.6.31       <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DirichletMultinomial       1.40.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  doParallel                 1.0.17       <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  doRNG                      1.8.6        <span style="color: #555555;">2023-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  dplyr                      1.1.0        <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  e1071                      1.7-12       <span style="color: #555555;">2022-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ellipsis                   0.3.2        <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  emmeans                    1.8.4-1      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  energy                     1.7-11       <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  estimability               1.4.1        <span style="color: #555555;">2022-08-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  evaluate                   0.20         <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Exact                      3.2          <span style="color: #555555;">2022-09-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  expm                       0.999-7      <span style="color: #555555;">2023-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  fansi                      1.0.4        <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  farver                     2.1.1        <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  fastmap                    1.1.0        <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  foreach                    1.5.2        <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  foreign                    0.8-83       <span style="color: #555555;">2022-09-28</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Formula                    1.2-4        <span style="color: #555555;">2020-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  fs                         1.6.0        <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  geneplotter                1.76.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  generics                   0.1.3        <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb               1.34.7       <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData           1.2.9        <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges              1.50.2       <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  getopt                     1.20.3       <span style="color: #555555;">2019-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm                 0.7.1        <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ggplot2                  * 3.4.0        <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ggrepel                    0.9.2        <span style="color: #555555;">2022-11-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gld                        2.6.6        <span style="color: #555555;">2022-10-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  glmnet                     4.1-6        <span style="color: #555555;">2022-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  glue                       1.6.2        <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gmp                        0.6-10       <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gplots                     3.1.3        <span style="color: #555555;">2022-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gridExtra                  2.3          <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gsl                        2.1-8        <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gtable                     0.3.1        <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gtools                     3.9.4        <span style="color: #555555;">2022-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  hash                       2.2.6.2      <span style="color: #555555;">2022-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  heatmaply                  1.4.2        <span style="color: #555555;">2023-01-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  highr                      0.10         <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Hmisc                      4.7-2        <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  htmlTable                  2.4.1        <span style="color: #555555;">2022-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  htmltools                  0.5.4        <span style="color: #555555;">2022-12-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets                1.6.1        <span style="color: #555555;">2023-01-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  httpuv                     1.6.8        <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  httr                       1.4.4        <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  igraph                     1.3.5        <span style="color: #555555;">2022-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  interp                     1.1-3        <span style="color: #555555;">2022-07-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  invgamma                   1.1          <span style="color: #555555;">2017-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  IRanges                    2.32.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  irlba                      2.3.5.1      <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  iterators                  1.0.14       <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  jpeg                       0.1-10       <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  jquerylib                  0.1.4        <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  jsonlite                   1.8.4        <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                   1.38.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  KernSmooth                 2.23-20      <span style="color: #555555;">2021-05-03</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  knitr                      1.42         <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  labeling                   0.4.2        <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  later                      1.3.0        <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lattice                    0.20-45      <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  latticeExtra               0.6-30       <span style="color: #555555;">2022-07-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lazyeval                   0.2.2        <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lefser                     1.8.0        <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  libcoin                    1.0-9        <span style="color: #555555;">2021-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lifecycle                  1.0.3        <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  limma                      3.54.1       <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  lme4                       1.1-31       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lmerTest                   3.1-3        <span style="color: #555555;">2020-10-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lmom                       2.9          <span style="color: #555555;">2022-05-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  locfit                     1.5-9.7      <span style="color: #555555;">2023-01-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  logging                    0.10-108     <span style="color: #555555;">2019-07-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lpSolveAPI                 5.5.2.0-17.9 <span style="color: #555555;">2022-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lpsymphony                 1.26.3       <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Bioconduc~</span></span></span>
<span><span class="co">#&gt;  Maaslin2                   1.12.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  magrittr                   2.0.3        <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  MASS                       7.3-58.1     <span style="color: #555555;">2022-08-03</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Matrix                     1.5-1        <span style="color: #555555;">2022-09-13</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics             1.10.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats                0.63.0       <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  memoise                    2.0.1        <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  metagenomeSeq              1.40.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  mgcv                       1.8-41       <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  mia                        1.6.0        <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  microbiome                 1.20.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  mime                       0.12         <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  miniUI                     0.1.1.1      <span style="color: #555555;">2018-05-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  minqa                      1.2.5        <span style="color: #555555;">2022-10-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  mixsqp                     0.3-48       <span style="color: #555555;">2022-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  modeltools                 0.2-23       <span style="color: #555555;">2020-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  multcomp                   1.4-20       <span style="color: #555555;">2022-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  MultiAssayExperiment       1.24.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  multtest                   2.54.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  munsell                    0.5.0        <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  mvtnorm                    1.1-3        <span style="color: #555555;">2021-10-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  NADA                       1.6-1.1      <span style="color: #555555;">2020-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  nlme                       3.1-160      <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  nloptr                     2.0.3        <span style="color: #555555;">2022-05-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  nnet                       7.3-18       <span style="color: #555555;">2022-09-28</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  numDeriv                   <span style="color: #BB00BB; font-weight: bold;">2016.8-1.1  </span> <span style="color: #555555;">2019-06-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  optparse                   1.7.3        <span style="color: #555555;">2022-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  pbapply                    1.7-0        <span style="color: #555555;">2023-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  pcaPP                      2.0-3        <span style="color: #555555;">2022-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  permute                    0.9-7        <span style="color: #555555;">2022-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  phyloseq                   1.42.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  pillar                     1.8.1        <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  pkgbuild                   1.4.0        <span style="color: #555555;">2022-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                  2.0.3        <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  pkgdown                    2.0.7        <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">any (@2.0.7)</span></span></span>
<span><span class="co">#&gt;  pkgload                    1.3.2        <span style="color: #555555;">2022-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  plotly                   * 4.10.1       <span style="color: #555555;">2022-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  plyr                       1.8.8        <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  png                        0.1-8        <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  prettyunits                1.1.1        <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  processx                   3.8.0        <span style="color: #555555;">2022-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  profvis                    0.3.7        <span style="color: #555555;">2020-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  promises                   1.2.0.1      <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  proxy                      0.4-27       <span style="color: #555555;">2022-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ps                         1.7.2        <span style="color: #555555;">2022-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  purrr                      1.0.1        <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  R6                         2.5.1        <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ragg                       1.2.5        <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rbibutils                  2.2.13       <span style="color: #555555;">2023-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer               1.1-3        <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Rcpp                       1.0.10       <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  RcppZiggurat               0.1.6        <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  RCurl                      1.98-1.10    <span style="color: #555555;">2023-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Rdpack                     2.4          <span style="color: #555555;">2022-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  readxl                     1.4.1        <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  registry                   0.5-1        <span style="color: #555555;">2019-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  remotes                    2.4.2        <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  reshape2                   1.4.4        <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Rfast                      2.0.6        <span style="color: #555555;">2022-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rhdf5                      2.42.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rhdf5filters               1.10.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib                   1.20.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rlang                      1.0.6        <span style="color: #555555;">2022-09-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                  2.20         <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Rmpfr                      0.9-0        <span style="color: #555555;">2023-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rngtools                   1.5.2        <span style="color: #555555;">2021-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  robustbase                 0.95-0       <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ROI                        1.0-0        <span style="color: #555555;">2020-08-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ROI.plugin.lpsolve         1.0-1        <span style="color: #555555;">2021-06-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rootSolve                  1.8.2.3      <span style="color: #555555;">2021-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rpart                      4.1.19       <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rprojroot                  2.0.3        <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  RSQLite                    2.2.20       <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rstatix                    0.7.1        <span style="color: #555555;">2022-11-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rstudioapi                 0.14         <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rsvd                       1.0.5        <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Rtsne                      0.16         <span style="color: #555555;">2022-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  S4Vectors                  0.36.1       <span style="color: #555555;">2022-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sandwich                   3.0-2        <span style="color: #555555;">2022-06-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  sass                       0.4.5        <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix               1.6.0        <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scales                     1.2.1        <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  scater                     1.26.1       <span style="color: #555555;">2022-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scuttle                    1.8.4        <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  seriation                  1.4.1        <span style="color: #555555;">2023-01-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  sessioninfo                1.2.2        <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  shape                      1.4.6        <span style="color: #555555;">2021-05-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  shiny                      1.7.4        <span style="color: #555555;">2022-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment       1.20.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  slam                       0.1-50       <span style="color: #555555;">2022-01-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  sparseMatrixStats          1.10.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  SQUAREM                    <span style="color: #BB00BB; font-weight: bold;">2021.1      </span> <span style="color: #555555;">2021-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  stringi                    1.7.12       <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  stringr                    1.5.0        <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment       1.28.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  survival                   3.4-0        <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  systemfonts                1.0.4        <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  testthat                   3.1.6        <span style="color: #555555;">2022-12-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  textshaping                0.3.6        <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  TH.data                    1.1-1        <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  tibble                     3.1.8        <span style="color: #555555;">2022-07-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  tidyr                      1.3.0        <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  tidyselect                 1.2.0        <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  tidytree                   0.4.2        <span style="color: #555555;">2022-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  treeio                     1.22.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  TreeSummarizedExperiment   2.6.0        <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  truncnorm                  1.0-8        <span style="color: #555555;">2018-02-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  trust                      0.1-8        <span style="color: #555555;">2020-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  TSP                        1.2-2        <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  UpSetR                     1.4.0        <span style="color: #555555;">2019-05-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  urlchecker                 1.0.1        <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  usethis                    2.1.6        <span style="color: #555555;">2022-05-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  utf8                       1.2.2        <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  vctrs                      0.5.2        <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  vegan                      2.6-4        <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  VGAM                       1.1-7        <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  vipor                      0.4.5        <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  viridis                    0.6.2        <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  viridisLite                0.4.1        <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  webshot                    0.5.4        <span style="color: #555555;">2022-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  withr                      2.5.0        <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Wrench                     1.16.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  xfun                       0.36         <span style="color: #555555;">2022-12-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  XML                        3.99-0.13    <span style="color: #555555;">2022-12-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  xtable                     1.8-4        <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  XVector                    0.38.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                       2.3.7        <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  yulab.utils                0.0.6        <span style="color: #555555;">2022-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  zCompositions              1.4.0-1      <span style="color: #555555;">2022-03-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                   1.44.0       <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  zoo                        1.8-11       <span style="color: #555555;">2022-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /home/runner/work/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /opt/R/4.2.2/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /opt/R/4.2.2/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Francesc Catala-Moll, Marc Noguera-Julian.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
