<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Start here if this is your first time usinc dar! You will learn about basic usage, steps, exploration and bakes.  
">
<title>Introduction to dar • dar</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to dar">
<meta property="og:description" content="Start here if this is your first time usinc dar! You will learn about basic usage, steps, exploration and bakes.  
">
<meta property="og:image" content="https://microbialgenomics-irsicaixaorg.github.io/dar/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dar</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/dar.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data_import.html">Data Import</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to dar</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar/blob/HEAD/vignettes/dar.Rmd" class="external-link"><code>vignettes/dar.Rmd</code></a></small>
      <div class="d-none name"><code>dar.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="an-example">An Example<a class="anchor" aria-label="anchor" href="#an-example"></a>
</h2>
<p>The package contains a data set from <a href="https://doi.org/10.1016/j.ebiom.2016.01.032" class="external-link">Noguera-Julian, M., et al. 2016</a> used to find the differential abundant microbial species among men who have sex with men (MSM) and non-MSM (hts). The data is stored inside an object of class <code>phyloseq</code>, the standard input format for creating recipes with dar. We will first load the data and inspect it:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/dar" class="external-link">dar</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metaHIV_phy"</span>, package <span class="op">=</span> <span class="st">"dar"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaHIV_phy</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 451 taxa and 156 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 156 samples by 3 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 451 taxa by 7 taxonomic ranks ]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="an-initial-recipe">An Initial Recipe<a class="anchor" aria-label="anchor" href="#an-initial-recipe"></a>
</h2>
<p>First, we will create a recipe object from the original data and then specify the processing and differential analysis steps.</p>
<p>Recipes can be created manually by sequentially adding roles to variables in a data set.</p>
<p>The easiest way to create the initial recipe is:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rec_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/recipe.html">recipe</a></span><span class="op">(</span><span class="va">metaHIV_phy</span>, var_info <span class="op">=</span> <span class="st">"RiskGroup2"</span>, tax_info <span class="op">=</span> <span class="st">"Species"</span><span class="op">)</span> </span>
<span><span class="va">rec_obj</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span></span></span></code></pre></div>
<p>The <code>var_info</code> argument corresponds to the variable to be considered in the modeling process and <code>tax_info</code> indicates the taxonomic level that will be used for the analyses.</p>
</div>
<div class="section level2">
<h2 id="preprocessing-steps">Preprocessing Steps<a class="anchor" aria-label="anchor" href="#preprocessing-steps"></a>
</h2>
<p>From here, preprocessing steps for some step X can be added sequentially in one of two ways:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>rec_obj <span class="ot">&lt;-</span> step_{X}(rec_obj, arguments)   </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="do">## or</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>rec_obj <span class="ot">&lt;-</span> rec_obj <span class="sc">%&gt;%</span> step_{X}(arguments)</span></code></pre></div>
<p>Note that all <code>step_ancom</code> and the other functions will always return updated recipes.</p>
<p>We have two types of steps, those in charge of processing (prepro) and those destined to define the methods of differential analysis (da).</p>
<p>The prepro steps are used to modify the data loaded into the recipe which will then be used for the da steps. The ‘dar’ package include 3 main preprocessing functionalities.</p>
<ul>
<li><p><code>step_subset_taxa</code>: Is used for subsetting the columns and values within the tax_table.</p></li>
<li><p><code>step_filter_taxa</code>: Is used for filtering OTUs from recipe objects.</p></li>
<li><p><code>step_rarefaction</code>: Is used to resample an OTU table such that all samples have the same library size.</p></li>
</ul>
<p>For our data, we can add an operation to preprocessing the data stored in the initial recpie. First, we will use <code>step_subset_taxa</code> to retain only Bacteria and Archaea OTUs from the Kingdom taxonomic level. We will then filter out OTUs where at least 3% of the samples have counts greater than 0.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rec_obj</span> <span class="op">&lt;-</span> <span class="va">rec_obj</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_subset_taxa.html">step_subset_taxa</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="st">'Kingdom %in% c("Bacteria", "Archaea")'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_filter_taxa.html">step_filter_taxa</a></span><span class="op">(</span>.f <span class="op">=</span> <span class="st">"function(x) sum(x &gt; 0) &gt;= (0.03 * length(x))"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">rec_obj</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Dutch_Baby_Pancake_(sometimes_called_a_German_pancake,_a_Bismarck,_or_a_Dutch_puff)</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_filter_taxa() <span style="color: #555555;">id = filter_taxa__Sou</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-analysis">Differential Analysis<a class="anchor" aria-label="anchor" href="#differential-analysis"></a>
</h2>
<p>Now that we have defined the preprocessing of the input data for all the da methods that will be used, we need to define them. For this introduction we will use <strong>metagenomeSeq</strong>, <strong>maaslin2</strong> and <strong>DESeq2</strong> methods with default parameters (those defined by the authors of each method).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rec_obj</span> <span class="op">&lt;-</span> <span class="va">rec_obj</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_deseq.html">step_deseq</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_metagenomeseq.html">step_metagenomeseq</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/step_maaslin.html">step_maaslin</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rec_obj</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Recipe</span> ──────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">451</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preporcessing steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_subset_taxa() <span style="color: #555555;">id = subset_taxa__Dutch_Baby_Pancake_(sometimes_called_a_German_pancake,_a_Bismarck,_or_a_Dutch_puff)</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_filter_taxa() <span style="color: #555555;">id = filter_taxa__Sou</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DA steps:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_deseq() <span style="color: #555555;">id = deseq__Bear_claw</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_metagenomeseq() <span style="color: #555555;">id = metagenomeseq__Samosa</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">◉</span> step_maaslin() <span style="color: #555555;">id = maaslin__Hot_water_crust_pastry</span></span></span></code></pre></div>
<p>The ‘dar’ package includes more da steps than those defined above. Below is the full list:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^step_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/ls.html" class="external-link">ls</a></span><span class="op">(</span><span class="st">"package:dar"</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"_new|_to_expr|filter|subset|rarefaction"</span>, <span class="va">.</span>, value <span class="op">=</span> <span class="cn">TRUE</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "step_aldex"         "step_ancom"         "step_corncob"      </span></span>
<span><span class="co">#&gt; [4] "step_deseq"         "step_lefse"         "step_maaslin"      </span></span>
<span><span class="co">#&gt; [7] "step_metagenomeseq" "step_wilcox"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prep">Prep<a class="anchor" aria-label="anchor" href="#prep"></a>
</h2>
<p>To ensure the reproducibility and consistency of the generated results, all the steps defined in the recipe are executed at the same time using the <code>prep</code> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep.html">prep</a></span><span class="op">(</span><span class="va">rec_obj</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">da_results</span></span>
<span><span class="co">#&gt; ── <span style="color: #0000BB;">DAR Results</span> ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Inputs:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> phyloseq object with <span style="color: #0000BB;">278</span> taxa and <span style="color: #0000BB;">156</span> samples </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> variable of interes <span style="color: #0000BB;">RiskGroup2</span> (class: character, levels: hts, msm, pwid) </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> taxonomic level <span style="color: #0000BB;">Species</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Results:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> deseq__Bear_claw <span style="color: #555555;">diff_taxa = 166</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> metagenomeseq__Samosa <span style="color: #555555;">diff_taxa = 237</span> </span></span>
<span><span class="co">#&gt;      <span style="color: #00BB00;">✔</span> maaslin__Hot_water_crust_pastry <span style="color: #555555;">diff_taxa = 205</span> </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      <span style="color: #0000BB;">ℹ</span> 100 taxa are present in all tested methods</span></span></code></pre></div>
<p>Note that the resulting object print shows information about the amount of differentially abundant OTUs in each of the methods, as well as the number of OTUs that have been detected by all methods (consensus).</p>
</div>
<div class="section level2">
<h2 id="bake-and-cool">Bake and cool<a class="anchor" aria-label="anchor" href="#bake-and-cool"></a>
</h2>
<p>Now that we have the results we need to extract them, however for this we first need to define a consensus strategy with the <code>bake</code>. For this example we are only interested in those OTUs detected as differentially abundant in the three methods used.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Number of used methods</span></span>
<span><span class="va">count</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/steps_ids.html">steps_ids</a></span><span class="op">(</span><span class="va">da_results</span>, type <span class="op">=</span> <span class="st">"da"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Define the bake </span></span>
<span><span class="va">da_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bake.html">bake</a></span><span class="op">(</span><span class="va">da_results</span>, count_cutoff <span class="op">=</span> <span class="va">count</span><span class="op">)</span></span></code></pre></div>
<p>Finally we can extract the table with the results using the <code>cool</code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cool.html">cool</a></span><span class="op">(</span><span class="va">da_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 2</span></span></span>
<span><span class="co">#&gt;    taxa_id taxa                        </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Otu_2   Methanosphaera_stadtmanae   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Otu_12  Bifidobacterium_animalis    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Otu_15  Bifidobacterium_catenulatum </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Otu_34  Olsenella_scatoligenes      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Otu_35  Collinsella_aerofaciens     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Otu_38  Enorma_massiliensis         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Otu_47  Bacteroides_cellulosilyticus</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Otu_48  Bacteroides_clarus          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Otu_63  Bacteroides_plebeius        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Otu_69  Bacteroides_sp_CAG_530      </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 90 more rows</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Francesc Catala-Moll, Marc Noguera-Julian.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
